// @angular/platform-server/platform-server.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
export { ɵAnimationRendererFactory as ɵAnimationRendererFactory_1 } from '@angular/platform-browser/animations';
export { ɵAnimationEngine as ɵAnimationEngine_18 } from '@angular/animations/browser';
import * as import0 from '@angular/core';
import * as import1 from '@angular/platform-server';
import * as import2 from '@angular/http';
import * as import3 from '@angular/common';
import * as import4 from '@angular/platform-browser';
import * as import5 from '@angular/platform-browser/animations';
import * as import6 from '@angular/animations/browser';
var ServerModuleInjector = (function (_super) {
    __extends(ServerModuleInjector, _super);
    function ServerModuleInjector(parent) {
        return _super.call(this, parent, [], []) || this;
    }
    Object.defineProperty(ServerModuleInjector.prototype, "_BrowserXhr_11", {
        get: function () {
            if ((this.__BrowserXhr_11 == null)) {
                (this.__BrowserXhr_11 = new import1.ɵc());
            }
            return this.__BrowserXhr_11;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_ResponseOptions_12", {
        get: function () {
            if ((this.__ResponseOptions_12 == null)) {
                (this.__ResponseOptions_12 = new import2.BaseResponseOptions());
            }
            return this.__ResponseOptions_12;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_XSRFStrategy_13", {
        get: function () {
            if ((this.__XSRFStrategy_13 == null)) {
                (this.__XSRFStrategy_13 = new import1.ɵd());
            }
            return this.__XSRFStrategy_13;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_XHRBackend_14", {
        get: function () {
            if ((this.__XHRBackend_14 == null)) {
                (this.__XHRBackend_14 = new import2.XHRBackend(this._BrowserXhr_11, this._ResponseOptions_12, this._XSRFStrategy_13));
            }
            return this.__XHRBackend_14;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_RequestOptions_15", {
        get: function () {
            if ((this.__RequestOptions_15 == null)) {
                (this.__RequestOptions_15 = new import2.BaseRequestOptions());
            }
            return this.__RequestOptions_15;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_Http_16", {
        get: function () {
            if ((this.__Http_16 == null)) {
                (this.__Http_16 = import1.ɵe(this._XHRBackend_14, this._RequestOptions_15));
            }
            return this.__Http_16;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_LOCALE_ID_17", {
        get: function () {
            if ((this.__LOCALE_ID_17 == null)) {
                (this.__LOCALE_ID_17 = import0.ɵn(this.parent.get(import0.LOCALE_ID, null)));
            }
            return this.__LOCALE_ID_17;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_NgLocalization_18", {
        get: function () {
            if ((this.__NgLocalization_18 == null)) {
                (this.__NgLocalization_18 = new import3.NgLocaleLocalization(this._LOCALE_ID_17));
            }
            return this.__NgLocalization_18;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_Compiler_19", {
        get: function () {
            if ((this.__Compiler_19 == null)) {
                (this.__Compiler_19 = new import0.Compiler());
            }
            return this.__Compiler_19;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_APP_ID_20", {
        get: function () {
            if ((this.__APP_ID_20 == null)) {
                (this.__APP_ID_20 = import0.ɵg());
            }
            return this.__APP_ID_20;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_IterableDiffers_21", {
        get: function () {
            if ((this.__IterableDiffers_21 == null)) {
                (this.__IterableDiffers_21 = import0.ɵl());
            }
            return this.__IterableDiffers_21;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_KeyValueDiffers_22", {
        get: function () {
            if ((this.__KeyValueDiffers_22 == null)) {
                (this.__KeyValueDiffers_22 = import0.ɵm());
            }
            return this.__KeyValueDiffers_22;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_DomSanitizer_23", {
        get: function () {
            if ((this.__DomSanitizer_23 == null)) {
                (this.__DomSanitizer_23 = new import4.ɵe(this.parent.get(import4.DOCUMENT)));
            }
            return this.__DomSanitizer_23;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_Sanitizer_24", {
        get: function () {
            if ((this.__Sanitizer_24 == null)) {
                (this.__Sanitizer_24 = this._DomSanitizer_23);
            }
            return this.__Sanitizer_24;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_HAMMER_GESTURE_CONFIG_25", {
        get: function () {
            if ((this.__HAMMER_GESTURE_CONFIG_25 == null)) {
                (this.__HAMMER_GESTURE_CONFIG_25 = new import4.HammerGestureConfig());
            }
            return this.__HAMMER_GESTURE_CONFIG_25;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_EVENT_MANAGER_PLUGINS_26", {
        get: function () {
            if ((this.__EVENT_MANAGER_PLUGINS_26 == null)) {
                (this.__EVENT_MANAGER_PLUGINS_26 = [
                    new import4.ɵDomEventsPlugin(this.parent.get(import4.DOCUMENT)),
                    new import4.ɵKeyEventsPlugin(this.parent.get(import4.DOCUMENT)),
                    new import4.ɵHammerGesturesPlugin(this.parent.get(import4.DOCUMENT), this._HAMMER_GESTURE_CONFIG_25)
                ]);
            }
            return this.__EVENT_MANAGER_PLUGINS_26;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_EventManager_27", {
        get: function () {
            if ((this.__EventManager_27 == null)) {
                (this.__EventManager_27 = new import4.EventManager(this._EVENT_MANAGER_PLUGINS_26, this.parent.get(import0.NgZone)));
            }
            return this.__EventManager_27;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_\u0275DomSharedStylesHost_28", {
        get: function () {
            if ((this.__ɵDomSharedStylesHost_28 == null)) {
                (this.__ɵDomSharedStylesHost_28 = new import4.ɵDomSharedStylesHost(this.parent.get(import4.DOCUMENT)));
            }
            return this.__ɵDomSharedStylesHost_28;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_\u0275DomRendererFactory2_29", {
        get: function () {
            if ((this.__ɵDomRendererFactory2_29 == null)) {
                (this.__ɵDomRendererFactory2_29 = new import4.ɵDomRendererFactory2(this._EventManager_27, this._ɵDomSharedStylesHost_28));
            }
            return this.__ɵDomRendererFactory2_29;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_\u0275b_30", {
        get: function () {
            if ((this.__ɵb_30 == null)) {
                (this.__ɵb_30 = new import1.ɵb(this.parent.get(import4.DOCUMENT), this.parent.get(import4.ɵTRANSITION_ID, null)));
            }
            return this.__ɵb_30;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_\u0275SharedStylesHost_31", {
        get: function () {
            if ((this.__ɵSharedStylesHost_31 == null)) {
                (this.__ɵSharedStylesHost_31 = this._ɵb_30);
            }
            return this.__ɵSharedStylesHost_31;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_\u0275ServerRendererFactory2_32", {
        get: function () {
            if ((this.__ɵServerRendererFactory2_32 == null)) {
                (this.__ɵServerRendererFactory2_32 = new import1.ɵServerRendererFactory2(this.parent.get(import0.NgZone), this.parent.get(import4.DOCUMENT), this._ɵSharedStylesHost_31));
            }
            return this.__ɵServerRendererFactory2_32;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_\u0275AnimationEngine_33", {
        get: function () {
            if ((this.__ɵAnimationEngine_33 == null)) {
                (this.__ɵAnimationEngine_33 = new import6.ɵNoopAnimationEngine());
            }
            return this.__ɵAnimationEngine_33;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_RendererFactory2_34", {
        get: function () {
            if ((this.__RendererFactory2_34 == null)) {
                (this.__RendererFactory2_34 = import1.ɵa(this._ɵServerRendererFactory2_32, this._ɵAnimationEngine_33, this.parent.get(import0.NgZone)));
            }
            return this.__RendererFactory2_34;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_Testability_35", {
        get: function () {
            if ((this.__Testability_35 == null)) {
                (this.__Testability_35 = null);
            }
            return this.__Testability_35;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_Meta_36", {
        get: function () {
            if ((this.__Meta_36 == null)) {
                (this.__Meta_36 = new import4.Meta(this.parent.get(import4.DOCUMENT)));
            }
            return this.__Meta_36;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ServerModuleInjector.prototype, "_Title_37", {
        get: function () {
            if ((this.__Title_37 == null)) {
                (this.__Title_37 = new import4.Title(this.parent.get(import4.DOCUMENT)));
            }
            return this.__Title_37;
        },
        enumerable: true,
        configurable: true
    });
    ServerModuleInjector.prototype.createInternal = function () {
        this._HttpModule_0 = new import2.HttpModule();
        this._CommonModule_1 = new import3.CommonModule();
        this._ErrorHandler_2 = import4.ɵa();
        this._APP_INITIALIZER_3 = [
            import0.ɵo,
            import4.ɵc(this.parent.get(import4.NgProbeToken, null), this.parent.get(import0.NgProbeToken, null))
        ];
        this._ApplicationInitStatus_4 = new import0.ApplicationInitStatus(this._APP_INITIALIZER_3);
        this._ɵf_5 = new import0.ɵf(this.parent.get(import0.NgZone), this.parent.get(import0.ɵConsole), this, this._ErrorHandler_2, this.componentFactoryResolver, this._ApplicationInitStatus_4);
        this._ApplicationRef_6 = this._ɵf_5;
        this._ApplicationModule_7 = new import0.ApplicationModule(this._ApplicationRef_6);
        this._BrowserModule_8 = new import4.BrowserModule(this.parent.get(import4.BrowserModule, null));
        this._NoopAnimationsModule_9 = new import5.NoopAnimationsModule();
        this._ServerModule_10 = new import1.ServerModule();
        return this._ServerModule_10;
    };
    ServerModuleInjector.prototype.getInternal = function (token, notFoundResult) {
        if ((token === import2.HttpModule)) {
            return this._HttpModule_0;
        }
        if ((token === import3.CommonModule)) {
            return this._CommonModule_1;
        }
        if ((token === import0.ErrorHandler)) {
            return this._ErrorHandler_2;
        }
        if ((token === import0.APP_INITIALIZER)) {
            return this._APP_INITIALIZER_3;
        }
        if ((token === import0.ApplicationInitStatus)) {
            return this._ApplicationInitStatus_4;
        }
        if ((token === import0.ɵf)) {
            return this._ɵf_5;
        }
        if ((token === import0.ApplicationRef)) {
            return this._ApplicationRef_6;
        }
        if ((token === import0.ApplicationModule)) {
            return this._ApplicationModule_7;
        }
        if ((token === import4.BrowserModule)) {
            return this._BrowserModule_8;
        }
        if ((token === import5.NoopAnimationsModule)) {
            return this._NoopAnimationsModule_9;
        }
        if ((token === import1.ServerModule)) {
            return this._ServerModule_10;
        }
        if ((token === import2.BrowserXhr)) {
            return this._BrowserXhr_11;
        }
        if ((token === import2.ResponseOptions)) {
            return this._ResponseOptions_12;
        }
        if ((token === import2.XSRFStrategy)) {
            return this._XSRFStrategy_13;
        }
        if ((token === import2.XHRBackend)) {
            return this._XHRBackend_14;
        }
        if ((token === import2.RequestOptions)) {
            return this._RequestOptions_15;
        }
        if ((token === import2.Http)) {
            return this._Http_16;
        }
        if ((token === import0.LOCALE_ID)) {
            return this._LOCALE_ID_17;
        }
        if ((token === import3.NgLocalization)) {
            return this._NgLocalization_18;
        }
        if ((token === import0.Compiler)) {
            return this._Compiler_19;
        }
        if ((token === import0.APP_ID)) {
            return this._APP_ID_20;
        }
        if ((token === import0.IterableDiffers)) {
            return this._IterableDiffers_21;
        }
        if ((token === import0.KeyValueDiffers)) {
            return this._KeyValueDiffers_22;
        }
        if ((token === import4.DomSanitizer)) {
            return this._DomSanitizer_23;
        }
        if ((token === import0.Sanitizer)) {
            return this._Sanitizer_24;
        }
        if ((token === import4.HAMMER_GESTURE_CONFIG)) {
            return this._HAMMER_GESTURE_CONFIG_25;
        }
        if ((token === import4.EVENT_MANAGER_PLUGINS)) {
            return this._EVENT_MANAGER_PLUGINS_26;
        }
        if ((token === import4.EventManager)) {
            return this._EventManager_27;
        }
        if ((token === import4.ɵDomSharedStylesHost)) {
            return this._ɵDomSharedStylesHost_28;
        }
        if ((token === import4.ɵDomRendererFactory2)) {
            return this._ɵDomRendererFactory2_29;
        }
        if ((token === import1.ɵb)) {
            return this._ɵb_30;
        }
        if ((token === import4.ɵSharedStylesHost)) {
            return this._ɵSharedStylesHost_31;
        }
        if ((token === import1.ɵServerRendererFactory2)) {
            return this._ɵServerRendererFactory2_32;
        }
        if ((token === import6.ɵAnimationEngine)) {
            return this._ɵAnimationEngine_33;
        }
        if ((token === import0.RendererFactory2)) {
            return this._RendererFactory2_34;
        }
        if ((token === import0.Testability)) {
            return this._Testability_35;
        }
        if ((token === import4.Meta)) {
            return this._Meta_36;
        }
        if ((token === import4.Title)) {
            return this._Title_37;
        }
        return notFoundResult;
    };
    ServerModuleInjector.prototype.destroyInternal = function () {
        this._ɵf_5.ngOnDestroy();
        (this.__ɵDomSharedStylesHost_28 && this._ɵDomSharedStylesHost_28.ngOnDestroy());
    };
    return ServerModuleInjector;
}(import0.ɵNgModuleInjector));
export var ServerModuleNgFactory = new import0.NgModuleFactory(ServerModuleInjector, import1.ServerModule);
