// @angular/material/core/selection/pseudo-checkbox/pseudo-checkbox.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
import * as import0 from './pseudo-checkbox.css.ngstyle';
import * as import1 from '@angular/core';
import * as import2 from '@angular/material/core/selection/pseudo-checkbox/pseudo-checkbox';
var styles_MdPseudoCheckbox = [import0.styles];
export var RenderType_MdPseudoCheckbox = import1.ɵcrt({
    encapsulation: 2,
    styles: styles_MdPseudoCheckbox,
    data: {}
});
export function View_MdPseudoCheckbox_0(l) {
    return import1.ɵvid(0, [], null, null);
}
function View_MdPseudoCheckbox_Host_0(l) {
    return import1.ɵvid(0, [
        (l()(), import1.ɵeld(0, null, null, 1, 'md-pseudo-checkbox', [], [
            [
                2,
                'mat-pseudo-checkbox',
                null
            ],
            [
                2,
                'mat-pseudo-checkbox-indeterminate',
                null
            ],
            [
                2,
                'mat-pseudo-checkbox-checked',
                null
            ],
            [
                2,
                'mat-pseudo-checkbox-disabled',
                null
            ]
        ], null, null, View_MdPseudoCheckbox_0, RenderType_MdPseudoCheckbox)),
        import1.ɵdid(24576, null, 0, import2.MdPseudoCheckbox, [
            import1.ElementRef,
            import1.Renderer
        ], null, null)
    ], null, function (ck, v) {
        var currVal_0 = true;
        var currVal_1 = (import1.ɵnov(v, 1).state === 'indeterminate');
        var currVal_2 = (import1.ɵnov(v, 1).state === 'checked');
        var currVal_3 = import1.ɵnov(v, 1).disabled;
        ck(v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3);
    });
}
export var MdPseudoCheckboxNgFactory = import1.ɵccf('md-pseudo-checkbox, mat-pseudo-checkbox', import2.MdPseudoCheckbox, View_MdPseudoCheckbox_Host_0, {
    state: 'state',
    disabled: 'disabled',
    color: 'color'
}, {}, []);
