// @angular/material/core/selection/pseudo-checkbox/pseudo-checkbox.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from './pseudo-checkbox.css.ngstyle';
import * as import1 from '@angular/core';
import * as import2 from '@angular/material/core/selection/pseudo-checkbox/pseudo-checkbox';
const styles_MdPseudoCheckbox:any[] = [import0.styles];
export const RenderType_MdPseudoCheckbox:import1.RendererType2 = import1.ɵcrt({
  encapsulation: 2,
  styles: styles_MdPseudoCheckbox,
  data: {}
}
);
export function View_MdPseudoCheckbox_0(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,([] as any[]),(null as any),(null as any));
}
function View_MdPseudoCheckbox_Host_0(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
    (l()(),import1.ɵeld(0,(null as any),(null as any),1,'md-pseudo-checkbox',([] as any[]),[
      [
        2,
        'mat-pseudo-checkbox',
        (null as any)
      ]
      ,
      [
        2,
        'mat-pseudo-checkbox-indeterminate',
        (null as any)
      ]
      ,
      [
        2,
        'mat-pseudo-checkbox-checked',
        (null as any)
      ]
      ,
      [
        2,
        'mat-pseudo-checkbox-disabled',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_MdPseudoCheckbox_0,RenderType_MdPseudoCheckbox)),
    import1.ɵdid(24576,(null as any),0,import2.MdPseudoCheckbox,[
      import1.ElementRef,
      import1.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = true;
    const currVal_1:any = (import1.ɵnov(v,1).state === 'indeterminate');
    const currVal_2:any = (import1.ɵnov(v,1).state === 'checked');
    const currVal_3:any = import1.ɵnov(v,1).disabled;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
export const MdPseudoCheckboxNgFactory:import1.ComponentFactory<import2.MdPseudoCheckbox> = import1.ɵccf('md-pseudo-checkbox, mat-pseudo-checkbox',import2.MdPseudoCheckbox,View_MdPseudoCheckbox_Host_0,{
  state: 'state',
  disabled: 'disabled',
  color: 'color'
}
,{},([] as any[]));
