// @angular/material/chips/chip-list.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/material/chips/chip-list';
import * as import2 from './chips.css.ngstyle';
class MdChipsModuleInjector extends import0.ɵNgModuleInjector<import1.MdChipsModule> {
  _MdChipsModule_0:import1.MdChipsModule;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  createInternal():import1.MdChipsModule {
    this._MdChipsModule_0 = new import1.MdChipsModule();
    return this._MdChipsModule_0;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import1.MdChipsModule)) { return this._MdChipsModule_0; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const MdChipsModuleNgFactory:import0.NgModuleFactory<import1.MdChipsModule> = new import0.NgModuleFactory<any>(MdChipsModuleInjector,import1.MdChipsModule);
const styles_MdChipList:any[] = [import2.styles];
export const RenderType_MdChipList:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_MdChipList,
  data: {}
}
);
export function View_MdChipList_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'md-chip-list-wrapper'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import0.ɵncd((null as any),0)
  ]
  ,(null as any),(null as any));
}
function View_MdChipList_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),2,'md-chip-list',[
      [
        'role',
        'listbox'
      ]
      ,
      [
        'tabindex',
        '0'
      ]

    ]
      ,[[
        2,
        'mat-chip-list',
        (null as any)
      ]
    ],[
      [
        (null as any),
        'focus'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('focus' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1).focus()) !== false);
        ad = (pd_0 && ad);
      }
      if (('keydown' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,1)._keydown($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },View_MdChipList_0,RenderType_MdChipList)),
    import0.ɵdid(548864,(null as any),1,import1.MdChipList,[import0.ElementRef],(null as any),(null as any)),
    import0.ɵqud(301989888,1,{chips: 1})
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = true;
    ck(v,0,0,currVal_0);
  });
}
export const MdChipListNgFactory:import0.ComponentFactory<import1.MdChipList> = import0.ɵccf('md-chip-list, mat-chip-list',import1.MdChipList,View_MdChipList_Host_0,{selectable: 'selectable'},{},['*']);
