// @angular/material/chips/chip.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
import * as import0 from '@angular/core';
import * as import1 from '@angular/material/chips/chip';
var styles_MdChip = [];
export var RenderType_MdChip = import0.ɵcrt({
    encapsulation: 2,
    styles: styles_MdChip,
    data: {}
});
export function View_MdChip_0(l) {
    return import0.ɵvid(0, [import0.ɵncd(null, 0)], null, null);
}
function View_MdChip_Host_0(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 1, 'md-basic-chip', [
            [
                'role',
                'option'
            ],
            [
                'tabindex',
                '-1'
            ]
        ], [
            [
                2,
                'mat-chip',
                null
            ],
            [
                2,
                'mat-chip-selected',
                null
            ],
            [
                1,
                'disabled',
                0
            ],
            [
                1,
                'aria-disabled',
                0
            ]
        ], [[
                null,
                'click'
            ]
        ], function (v, en, $event) {
            var ad = true;
            if (('click' === en)) {
                var pd_0 = (import0.ɵnov(v, 1)._handleClick($event) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, View_MdChip_0, RenderType_MdChip)),
        import0.ɵdid(122880, null, 0, import1.MdChip, [
            import0.Renderer,
            import0.ElementRef
        ], null, null)
    ], function (ck, v) {
        ck(v, 1, 0);
    }, function (ck, v) {
        var currVal_0 = true;
        var currVal_1 = import0.ɵnov(v, 1).selected;
        var currVal_2 = import0.ɵnov(v, 1).disabled;
        var currVal_3 = import0.ɵnov(v, 1)._isAriaDisabled;
        ck(v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3);
    });
}
export var MdChipNgFactory = import0.ɵccf('md-basic-chip, [md-basic-chip], md-chip, [md-chip],\n             mat-basic-chip, [mat-basic-chip], mat-chip, [mat-chip]', import1.MdChip, View_MdChip_Host_0, {
    disabled: 'disabled',
    selected: 'selected',
    color: 'color'
}, {
    select: 'select',
    deselect: 'deselect',
    destroy: 'destroy'
}, ['*']);
