// @angular/material/chips/chip.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/material/chips/chip';
const styles_MdChip:any[] = ([] as any[]);
export const RenderType_MdChip:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_MdChip,
  data: {}
}
);
export function View_MdChip_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[import0.ɵncd((null as any),0)],(null as any),(null as any));
}
function View_MdChip_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'md-basic-chip',[
      [
        'role',
        'option'
      ]
      ,
      [
        'tabindex',
        '-1'
      ]

    ]
    ,[
      [
        2,
        'mat-chip',
        (null as any)
      ]
      ,
      [
        2,
        'mat-chip-selected',
        (null as any)
      ]
      ,
      [
        1,
        'disabled',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1)._handleClick($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_MdChip_0,RenderType_MdChip)),
    import0.ɵdid(122880,(null as any),0,import1.MdChip,[
      import0.Renderer,
      import0.ElementRef
    ]
    ,(null as any),(null as any))
  ]
  ,(ck,v) => {
    ck(v,1,0);
  },(ck,v) => {
    const currVal_0:any = true;
    const currVal_1:any = import0.ɵnov(v,1).selected;
    const currVal_2:any = import0.ɵnov(v,1).disabled;
    const currVal_3:any = import0.ɵnov(v,1)._isAriaDisabled;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
export const MdChipNgFactory:import0.ComponentFactory<import1.MdChip> = import0.ɵccf('md-basic-chip, [md-basic-chip], md-chip, [md-chip],\n             mat-basic-chip, [mat-basic-chip], mat-chip, [mat-chip]',import1.MdChip,View_MdChip_Host_0,{
  disabled: 'disabled',
  selected: 'selected',
  color: 'color'
}
,{
  select: 'select',
  deselect: 'deselect',
  destroy: 'destroy'
}
,['*']);
