// @angular/material/select/select.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
import * as import0 from './select.css.ngstyle';
import * as import1 from '@angular/core';
import * as import2 from '@angular/material/select/select';
import * as import3 from '@angular/material/core/overlay/overlay-directives';
import * as import4 from '@angular/common';
import * as import5 from '@angular/material/core/overlay/overlay';
import * as import6 from '@angular/material/core/rtl/dir';
import * as import7 from '@angular/material/core/overlay/position/viewport-ruler';
var styles_MdSelect = [import0.styles];
export var RenderType_MdSelect = import1.ɵcrt({
    encapsulation: 2,
    styles: styles_MdSelect,
    data: { 'animation': [
            {
                name: 'transformPlaceholder',
                definitions: [
                    {
                        type: 0,
                        name: 'floating-ltr',
                        styles: {
                            type: 6,
                            styles: {
                                top: '-22px',
                                left: '-2px',
                                transform: 'scale(0.75)'
                            }
                        }
                    },
                    {
                        type: 0,
                        name: 'floating-rtl',
                        styles: {
                            type: 6,
                            styles: {
                                top: '-22px',
                                left: '2px',
                                transform: 'scale(0.75)'
                            }
                        }
                    },
                    {
                        type: 1,
                        expr: '* => *',
                        animation: {
                            type: 4,
                            styles: null,
                            timings: '400ms cubic-bezier(0.25, 0.8, 0.25, 1)'
                        }
                    }
                ]
            },
            {
                name: 'transformPanel',
                definitions: [
                    {
                        type: 0,
                        name: 'showing',
                        styles: {
                            type: 6,
                            styles: {
                                opacity: 1,
                                minWidth: 'calc(100% + 32px)',
                                transform: 'translate3d(0,0,0) scaleY(1)'
                            }
                        }
                    },
                    {
                        type: 1,
                        expr: 'void => *',
                        animation: [
                            {
                                type: 6,
                                styles: {
                                    opacity: 0,
                                    minWidth: '100%',
                                    transform: 'translate3d(0, 0, 0) scaleY(0)'
                                }
                            },
                            {
                                type: 4,
                                styles: null,
                                timings: '150ms cubic-bezier(0.25, 0.8, 0.25, 1)'
                            }
                        ]
                    },
                    {
                        type: 1,
                        expr: '* => void',
                        animation: [{
                                type: 4,
                                styles: {
                                    type: 6,
                                    styles: { opacity: 0 }
                                },
                                timings: '250ms 100ms linear'
                            }
                        ]
                    }
                ]
            },
            {
                name: 'fadeInContent',
                definitions: [
                    {
                        type: 0,
                        name: 'showing',
                        styles: {
                            type: 6,
                            styles: { opacity: 1 }
                        }
                    },
                    {
                        type: 1,
                        expr: 'void => showing',
                        animation: [
                            {
                                type: 6,
                                styles: { opacity: 0 }
                            },
                            {
                                type: 4,
                                styles: null,
                                timings: '150ms 100ms cubic-bezier(0.55, 0, 0.55, 0.2)'
                            }
                        ]
                    }
                ]
            }
        ]
    }
});
function View_MdSelect_1(l) {
    return import1.ɵvid(0, [
        (l()(), import1.ɵeld(0, null, null, 3, 'span', [[
                'class',
                'mat-select-value'
            ]
        ], null, null, null, null, null)),
        (l()(), import1.ɵeld(0, null, null, 1, 'span', [[
                'class',
                'mat-select-value-text'
            ]
        ], null, null, null, null, null)),
        (l()(), import1.ɵted(null, [
            '',
            ''
        ])),
        (l()(), import1.ɵted(null, [' ']))
    ], null, function (ck, v) {
        var co = v.component;
        var currVal_0 = ((co.selected == null) ? null : co.selected.viewValue);
        ck(v, 2, 0, currVal_0);
    });
}
function View_MdSelect_2(l) {
    return import1.ɵvid(0, [
        (l()(), import1.ɵeld(0, null, null, 2, 'div', [[
                'class',
                'mat-select-panel'
            ]
        ], [
            [
                24,
                '@transformPanel',
                0
            ],
            [
                4,
                'transformOrigin',
                null
            ],
            [
                2,
                'mat-select-panel-done-animating',
                null
            ]
        ], [
            [
                null,
                '@transformPanel.done'
            ],
            [
                null,
                'keydown'
            ]
        ], function (v, en, $event) {
            var ad = true;
            var co = v.component;
            if (('@transformPanel.done' === en)) {
                var pd_0 = (co._onPanelDone() !== false);
                ad = (pd_0 && ad);
            }
            if (('keydown' === en)) {
                var pd_1 = (co._keyManager.onKeydown($event) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)),
        (l()(), import1.ɵeld(0, null, null, 1, 'div', [[
                'class',
                'mat-select-content'
            ]
        ], [[
                24,
                '@fadeInContent',
                0
            ]
        ], [[
                null,
                '@fadeInContent.done'
            ]
        ], function (v, en, $event) {
            var ad = true;
            var co = v.component;
            if (('@fadeInContent.done' === en)) {
                var pd_0 = (co._onFadeInDone() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)),
        import1.ɵncd(null, 0)
    ], null, function (ck, v) {
        var co = v.component;
        var currVal_0 = 'showing';
        var currVal_1 = co._transformOrigin;
        var currVal_2 = co._panelDoneAnimating;
        ck(v, 0, 0, currVal_0, currVal_1, currVal_2);
        var currVal_3 = 'showing';
        ck(v, 1, 0, currVal_3);
    });
}
export function View_MdSelect_0(l) {
    return import1.ɵvid(0, [
        import1.ɵqud(201326592, 1, { trigger: 0 }),
        import1.ɵqud(201326592, 2, { overlayDir: 0 }),
        (l()(), import1.ɵeld(0, [
            [
                1,
                0
            ],
            [
                'trigger',
                1
            ]
        ], null, 8, 'div', [
            [
                'cdk-overlay-origin',
                ''
            ],
            [
                'class',
                'mat-select-trigger'
            ]
        ], null, [[
                null,
                'click'
            ]
        ], function (v, en, $event) {
            var ad = true;
            var co = v.component;
            if (('click' === en)) {
                var pd_0 = (co.toggle() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)),
        import1.ɵdid(8192, [[
                'origin',
                4
            ]
        ], 0, import3.OverlayOrigin, [import1.ElementRef], null, null),
        (l()(), import1.ɵeld(0, null, null, 1, 'span', [[
                'class',
                'mat-select-placeholder'
            ]
        ], [
            [
                2,
                'mat-floating-placeholder',
                null
            ],
            [
                24,
                '@transformPlaceholder',
                0
            ],
            [
                4,
                'width',
                'px'
            ]
        ], null, null, null, null)),
        (l()(), import1.ɵted(null, [
            '',
            ' '
        ])),
        (l()(), import1.ɵand(8388608, null, null, 1, null, View_MdSelect_1)),
        import1.ɵdid(8192, null, 0, import4.NgIf, [
            import1.ViewContainerRef,
            import1.TemplateRef
        ], { ngIf: [
                0,
                'ngIf'
            ]
        }, null),
        (l()(), import1.ɵeld(0, null, null, 0, 'span', [[
                'class',
                'mat-select-arrow'
            ]
        ], null, null, null, null, null)),
        (l()(), import1.ɵted(null, [' '])),
        (l()(), import1.ɵeld(0, null, null, 0, 'span', [[
                'class',
                'mat-select-underline'
            ]
        ], null, null, null, null, null)),
        (l()(), import1.ɵand(8388608, null, null, 1, function (v, en, $event) {
            var ad = true;
            var co = v.component;
            if (('backdropClick' === en)) {
                var pd_0 = (co.close() !== false);
                ad = (pd_0 && ad);
            }
            if (('attach' === en)) {
                var pd_1 = (co._setScrollTop() !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, View_MdSelect_2)),
        import1.ɵdid(73728, [[
                2,
                4
            ]
        ], 0, import3.ConnectedOverlayDirective, [
            import5.Overlay,
            import1.TemplateRef,
            import1.ViewContainerRef,
            [
                2,
                import6.Dir
            ]
        ], {
            origin: [
                0,
                'origin'
            ],
            positions: [
                1,
                'positions'
            ],
            offsetX: [
                2,
                'offsetX'
            ],
            offsetY: [
                3,
                'offsetY'
            ],
            minWidth: [
                4,
                'minWidth'
            ],
            backdropClass: [
                5,
                'backdropClass'
            ],
            hasBackdrop: [
                6,
                'hasBackdrop'
            ],
            open: [
                7,
                'open'
            ]
        }, {
            backdropClick: 'backdropClick',
            attach: 'attach'
        })
    ], function (ck, v) {
        var co = v.component;
        var currVal_4 = co.selected;
        ck(v, 7, 0, currVal_4);
        var currVal_5 = import1.ɵnov(v, 3);
        var currVal_6 = co._positions;
        var currVal_7 = co._offsetX;
        var currVal_8 = co._offsetY;
        var currVal_9 = co._triggerWidth;
        var currVal_10 = 'cdk-overlay-transparent-backdrop';
        var currVal_11 = '';
        var currVal_12 = co.panelOpen;
        ck(v, 12, 0, currVal_5, currVal_6, currVal_7, currVal_8, currVal_9, currVal_10, currVal_11, currVal_12);
    }, function (ck, v) {
        var co = v.component;
        var currVal_0 = co.selected;
        var currVal_1 = co._placeholderState;
        var currVal_2 = co._selectedValueWidth;
        ck(v, 4, 0, currVal_0, currVal_1, currVal_2);
        var currVal_3 = co.placeholder;
        ck(v, 5, 0, currVal_3);
    });
}
function View_MdSelect_Host_0(l) {
    return import1.ɵvid(0, [
        (l()(), import1.ɵeld(0, null, null, 2, 'md-select', [[
                'role',
                'listbox'
            ]
        ], [
            [
                1,
                'tabindex',
                0
            ],
            [
                1,
                'aria-label',
                0
            ],
            [
                1,
                'aria-required',
                0
            ],
            [
                1,
                'aria-disabled',
                0
            ],
            [
                1,
                'aria-invalid',
                0
            ],
            [
                1,
                'aria-owns',
                0
            ],
            [
                2,
                'mat-select-disabled',
                null
            ],
            [
                2,
                'mat-select',
                null
            ]
        ], [
            [
                null,
                'keydown'
            ],
            [
                null,
                'blur'
            ]
        ], function (v, en, $event) {
            var ad = true;
            if (('keydown' === en)) {
                var pd_0 = (import1.ɵnov(v, 1)._handleKeydown($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('blur' === en)) {
                var pd_1 = (import1.ɵnov(v, 1)._onBlur() !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, View_MdSelect_0, RenderType_MdSelect)),
        import1.ɵdid(614400, null, 1, import2.MdSelect, [
            import1.ElementRef,
            import1.Renderer,
            import7.ViewportRuler,
            import1.ChangeDetectorRef,
            [
                2,
                import6.Dir
            ],
            [
                8,
                null
            ]
        ], null, null),
        import1.ɵqud(301989888, 1, { options: 1 })
    ], null, function (ck, v) {
        var currVal_0 = import1.ɵnov(v, 1)._getTabIndex();
        var currVal_1 = import1.ɵnov(v, 1).placeholder;
        var currVal_2 = import1.ɵnov(v, 1).required.toString();
        var currVal_3 = import1.ɵnov(v, 1).disabled.toString();
        var currVal_4 = (((import1.ɵnov(v, 1)._control == null) ? null : import1.ɵnov(v, 1)._control.invalid) || 'false');
        var currVal_5 = import1.ɵnov(v, 1)._optionIds;
        var currVal_6 = import1.ɵnov(v, 1).disabled;
        var currVal_7 = true;
        ck(v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4, currVal_5, currVal_6, currVal_7);
    });
}
export var MdSelectNgFactory = import1.ɵccf('md-select, mat-select', import2.MdSelect, View_MdSelect_Host_0, {
    placeholder: 'placeholder',
    disabled: 'disabled',
    required: 'required'
}, {
    onOpen: 'onOpen',
    onClose: 'onClose',
    change: 'change'
}, ['*']);
