// @angular/material/select/select.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from './select.css.ngstyle';
import * as import1 from '@angular/core';
import * as import2 from '@angular/material/select/select';
import * as import3 from '@angular/material/core/overlay/overlay-directives';
import * as import4 from '@angular/common';
import * as import5 from '@angular/material/core/overlay/overlay';
import * as import6 from '@angular/material/core/rtl/dir';
import * as import7 from '@angular/material/core/overlay/position/viewport-ruler';
const styles_MdSelect:any[] = [import0.styles];
export const RenderType_MdSelect:import1.RendererType2 = import1.ɵcrt({
  encapsulation: 2,
  styles: styles_MdSelect,
    data: {'animation': [
      {
        name: 'transformPlaceholder',
        definitions: [
          {
            type: 0,
            name: 'floating-ltr',
            styles: {
              type: 6,
              styles: {
                top: '-22px',
                left: '-2px',
                transform: 'scale(0.75)'
              }

            }

          }
          ,
          {
            type: 0,
            name: 'floating-rtl',
            styles: {
              type: 6,
              styles: {
                top: '-22px',
                left: '2px',
                transform: 'scale(0.75)'
              }

            }

          }
          ,
          {
            type: 1,
            expr: '* => *',
            animation: {
              type: 4,
              styles: (null as any),
              timings: '400ms cubic-bezier(0.25, 0.8, 0.25, 1)'
            }

          }

        ]

      }
      ,
      {
        name: 'transformPanel',
        definitions: [
          {
            type: 0,
            name: 'showing',
            styles: {
              type: 6,
              styles: {
                opacity: 1,
                minWidth: 'calc(100% + 32px)',
                transform: 'translate3d(0,0,0) scaleY(1)'
              }

            }

          }
          ,
          {
            type: 1,
            expr: 'void => *',
            animation: [
              {
                type: 6,
                styles: {
                  opacity: 0,
                  minWidth: '100%',
                  transform: 'translate3d(0, 0, 0) scaleY(0)'
                }

              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: '150ms cubic-bezier(0.25, 0.8, 0.25, 1)'
              }

            ]

          }
          ,
          {
            type: 1,
            expr: '* => void',
              animation: [{
                type: 4,
                styles: {
                  type: 6,
                  styles: {opacity: 0}
                }
                ,
                timings: '250ms 100ms linear'
              }
            ]
          }

        ]

      }
      ,
      {
        name: 'fadeInContent',
        definitions: [
          {
            type: 0,
            name: 'showing',
            styles: {
              type: 6,
              styles: {opacity: 1}
            }

          }
          ,
          {
            type: 1,
            expr: 'void => showing',
            animation: [
              {
                type: 6,
                styles: {opacity: 0}
              }
              ,
              {
                type: 4,
                styles: (null as any),
                timings: '150ms 100ms cubic-bezier(0.55, 0, 0.55, 0.2)'
              }

            ]

          }

        ]

      }

    ]
  }
}
);
function View_MdSelect_1(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
      (l()(),import1.ɵeld(0,(null as any),(null as any),3,'span',[[
        'class',
        'mat-select-value'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import1.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'mat-select-value-text'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import1.ɵted((null as any),[
      '',
      ''
    ]
    )),
    (l()(),import1.ɵted((null as any),[' ']))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = ((co.selected == null)? (null as any): co.selected.viewValue);
    ck(v,2,0,currVal_0);
  });
}
function View_MdSelect_2(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
      (l()(),import1.ɵeld(0,(null as any),(null as any),2,'div',[[
        'class',
        'mat-select-panel'
      ]
    ],[
      [
        24,
        '@transformPanel',
        0
      ]
      ,
      [
        4,
        'transformOrigin',
        (null as any)
      ]
      ,
      [
        2,
        'mat-select-panel-done-animating',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        '@transformPanel.done'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('@transformPanel.done' === en)) {
        const pd_0:any = ((<any>co._onPanelDone()) !== false);
        ad = (pd_0 && ad);
      }
      if (('keydown' === en)) {
        const pd_1:any = ((<any>co._keyManager.onKeydown($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },(null as any),(null as any))),
      (l()(),import1.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'mat-select-content'
      ]
      ],[[
        24,
        '@fadeInContent',
        0
      ]
      ],[[
        (null as any),
        '@fadeInContent.done'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:any = v.component;
      if (('@fadeInContent.done' === en)) {
        const pd_0:any = ((<any>co._onFadeInDone()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import1.ɵncd((null as any),0)
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = 'showing';
    const currVal_1:any = co._transformOrigin;
    const currVal_2:any = co._panelDoneAnimating;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
    const currVal_3:any = 'showing';
    ck(v,1,0,currVal_3);
  });
}
export function View_MdSelect_0(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
    import1.ɵqud(201326592,1,{trigger: 0}),
    import1.ɵqud(201326592,2,{overlayDir: 0}),
    (l()(),import1.ɵeld(0,[
      [
        1,
        0
      ]
      ,
      [
        'trigger',
        1
      ]

    ]
    ,(null as any),8,'div',[
      [
        'cdk-overlay-origin',
        ''
      ]
      ,
      [
        'class',
        'mat-select-trigger'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import2.MdSelect = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co.toggle()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
      import1.ɵdid(8192,[[
        'origin',
        4
      ]
    ],0,import3.OverlayOrigin,[import1.ElementRef],(null as any),(null as any)),
      (l()(),import1.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'mat-select-placeholder'
      ]
    ],[
      [
        2,
        'mat-floating-placeholder',
        (null as any)
      ]
      ,
      [
        24,
        '@transformPlaceholder',
        0
      ]
      ,
      [
        4,
        'width',
        'px'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import1.ɵted((null as any),[
      '',
      ' '
    ]
    )),
    (l()(),import1.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MdSelect_1)),
    import1.ɵdid(8192,(null as any),0,import4.NgIf,[
      import1.ViewContainerRef,
      import1.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
      (l()(),import1.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'mat-select-arrow'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import1.ɵted((null as any),[' '])),
      (l()(),import1.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'mat-select-underline'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import1.ɵand(8388608,(null as any),(null as any),1,(v,en,$event) => {
      var ad:boolean = true;
      var co:import2.MdSelect = v.component;
      if (('backdropClick' === en)) {
        const pd_0:any = ((<any>co.close()) !== false);
        ad = (pd_0 && ad);
      }
      if (('attach' === en)) {
        const pd_1:any = ((<any>co._setScrollTop()) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },View_MdSelect_2)),
      import1.ɵdid(73728,[[
        2,
        4
      ]
    ],0,import3.ConnectedOverlayDirective,[
      import5.Overlay,
      import1.TemplateRef,
      import1.ViewContainerRef,
      [
        2,
        import6.Dir
      ]

    ]
    ,{
      origin: [
        0,
        'origin'
      ]
      ,
      positions: [
        1,
        'positions'
      ]
      ,
      offsetX: [
        2,
        'offsetX'
      ]
      ,
      offsetY: [
        3,
        'offsetY'
      ]
      ,
      minWidth: [
        4,
        'minWidth'
      ]
      ,
      backdropClass: [
        5,
        'backdropClass'
      ]
      ,
      hasBackdrop: [
        6,
        'hasBackdrop'
      ]
      ,
      open: [
        7,
        'open'
      ]

    }
    ,{
      backdropClick: 'backdropClick',
      attach: 'attach'
    }
    )
  ]
  ,(ck,v) => {
    var co:import2.MdSelect = v.component;
    const currVal_4:any = co.selected;
    ck(v,7,0,currVal_4);
    const currVal_5:any = import1.ɵnov(v,3);
    const currVal_6:any = co._positions;
    const currVal_7:any = co._offsetX;
    const currVal_8:any = co._offsetY;
    const currVal_9:any = co._triggerWidth;
    const currVal_10:any = 'cdk-overlay-transparent-backdrop';
    const currVal_11:any = '';
    const currVal_12:any = co.panelOpen;
    ck(v,12,0,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9,currVal_10,currVal_11,currVal_12);
  },(ck,v) => {
    var co:import2.MdSelect = v.component;
    const currVal_0:any = co.selected;
    const currVal_1:any = co._placeholderState;
    const currVal_2:any = co._selectedValueWidth;
    ck(v,4,0,currVal_0,currVal_1,currVal_2);
    const currVal_3:any = co.placeholder;
    ck(v,5,0,currVal_3);
  });
}
function View_MdSelect_Host_0(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
      (l()(),import1.ɵeld(0,(null as any),(null as any),2,'md-select',[[
        'role',
        'listbox'
      ]
    ],[
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        1,
        'aria-label',
        0
      ]
      ,
      [
        1,
        'aria-required',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]
      ,
      [
        1,
        'aria-invalid',
        0
      ]
      ,
      [
        1,
        'aria-owns',
        0
      ]
      ,
      [
        2,
        'mat-select-disabled',
        (null as any)
      ]
      ,
      [
        2,
        'mat-select',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'keydown'
      ]
      ,
      [
        (null as any),
        'blur'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('keydown' === en)) {
        const pd_0:any = ((<any>import1.ɵnov(v,1)._handleKeydown($event)) !== false);
        ad = (pd_0 && ad);
      }
      if (('blur' === en)) {
        const pd_1:any = ((<any>import1.ɵnov(v,1)._onBlur()) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },View_MdSelect_0,RenderType_MdSelect)),
    import1.ɵdid(614400,(null as any),1,import2.MdSelect,[
      import1.ElementRef,
      import1.Renderer,
      import7.ViewportRuler,
      import1.ChangeDetectorRef,
      [
        2,
        import6.Dir
      ]
      ,
      [
        8,
        (null as any)
      ]

    ]
    ,(null as any),(null as any)),
    import1.ɵqud(301989888,1,{options: 1})
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import1.ɵnov(v,1)._getTabIndex();
    const currVal_1:any = import1.ɵnov(v,1).placeholder;
    const currVal_2:any = import1.ɵnov(v,1).required.toString();
    const currVal_3:any = import1.ɵnov(v,1).disabled.toString();
    const currVal_4:any = (((import1.ɵnov(v,1)._control == null)? (null as any): import1.ɵnov(v,1)._control.invalid) || 'false');
    const currVal_5:any = import1.ɵnov(v,1)._optionIds;
    const currVal_6:any = import1.ɵnov(v,1).disabled;
    const currVal_7:any = true;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7);
  });
}
export const MdSelectNgFactory:import1.ComponentFactory<import2.MdSelect> = import1.ɵccf('md-select, mat-select',import2.MdSelect,View_MdSelect_Host_0,{
  placeholder: 'placeholder',
  disabled: 'disabled',
  required: 'required'
}
,{
  onOpen: 'onOpen',
  onClose: 'onClose',
  change: 'change'
}
,['*']);
