// @angular/material/core/compatibility/compatibility.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import * as import0 from '@angular/core';
import * as import1 from '@angular/material/core/compatibility/compatibility';
var CompatibilityModuleInjector = (function (_super) {
    __extends(CompatibilityModuleInjector, _super);
    function CompatibilityModuleInjector(parent) {
        return _super.call(this, parent, [], []) || this;
    }
    CompatibilityModuleInjector.prototype.createInternal = function () {
        this._CompatibilityModule_0 = new import1.CompatibilityModule();
        return this._CompatibilityModule_0;
    };
    CompatibilityModuleInjector.prototype.getInternal = function (token, notFoundResult) {
        if ((token === import1.CompatibilityModule)) {
            return this._CompatibilityModule_0;
        }
        return notFoundResult;
    };
    CompatibilityModuleInjector.prototype.destroyInternal = function () {
    };
    return CompatibilityModuleInjector;
}(import0.ɵNgModuleInjector));
export var CompatibilityModuleNgFactory = new import0.NgModuleFactory(CompatibilityModuleInjector, import1.CompatibilityModule);
var NoConflictStyleCompatibilityModeInjector = (function (_super) {
    __extends(NoConflictStyleCompatibilityModeInjector, _super);
    function NoConflictStyleCompatibilityModeInjector(parent) {
        return _super.call(this, parent, [], []) || this;
    }
    NoConflictStyleCompatibilityModeInjector.prototype.createInternal = function () {
        this._NoConflictStyleCompatibilityMode_0 = new import1.NoConflictStyleCompatibilityMode();
        this._MATERIAL_COMPATIBILITY_MODE_1 = true;
        return this._NoConflictStyleCompatibilityMode_0;
    };
    NoConflictStyleCompatibilityModeInjector.prototype.getInternal = function (token, notFoundResult) {
        if ((token === import1.NoConflictStyleCompatibilityMode)) {
            return this._NoConflictStyleCompatibilityMode_0;
        }
        if ((token === import1.MATERIAL_COMPATIBILITY_MODE)) {
            return this._MATERIAL_COMPATIBILITY_MODE_1;
        }
        return notFoundResult;
    };
    NoConflictStyleCompatibilityModeInjector.prototype.destroyInternal = function () {
    };
    return NoConflictStyleCompatibilityModeInjector;
}(import0.ɵNgModuleInjector));
export var NoConflictStyleCompatibilityModeNgFactory = new import0.NgModuleFactory(NoConflictStyleCompatibilityModeInjector, import1.NoConflictStyleCompatibilityMode);
