// @angular/material/core/compatibility/compatibility.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/material/core/compatibility/compatibility';
class CompatibilityModuleInjector extends import0.ɵNgModuleInjector<import1.CompatibilityModule> {
  _CompatibilityModule_0:import1.CompatibilityModule;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  createInternal():import1.CompatibilityModule {
    this._CompatibilityModule_0 = new import1.CompatibilityModule();
    return this._CompatibilityModule_0;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import1.CompatibilityModule)) { return this._CompatibilityModule_0; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const CompatibilityModuleNgFactory:import0.NgModuleFactory<import1.CompatibilityModule> = new import0.NgModuleFactory<any>(CompatibilityModuleInjector,import1.CompatibilityModule);
class NoConflictStyleCompatibilityModeInjector extends import0.ɵNgModuleInjector<import1.NoConflictStyleCompatibilityMode> {
  _NoConflictStyleCompatibilityMode_0:import1.NoConflictStyleCompatibilityMode;
  _MATERIAL_COMPATIBILITY_MODE_1:any;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  createInternal():import1.NoConflictStyleCompatibilityMode {
    this._NoConflictStyleCompatibilityMode_0 = new import1.NoConflictStyleCompatibilityMode();
    this._MATERIAL_COMPATIBILITY_MODE_1 = true;
    return this._NoConflictStyleCompatibilityMode_0;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import1.NoConflictStyleCompatibilityMode)) { return this._NoConflictStyleCompatibilityMode_0; }
    if ((token === import1.MATERIAL_COMPATIBILITY_MODE)) { return this._MATERIAL_COMPATIBILITY_MODE_1; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const NoConflictStyleCompatibilityModeNgFactory:import0.NgModuleFactory<import1.NoConflictStyleCompatibilityMode> = new import0.NgModuleFactory<any>(NoConflictStyleCompatibilityModeInjector,import1.NoConflictStyleCompatibilityMode);
