// @angular/material/input/input-container.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from './input-container.css.ngstyle';
import * as import1 from '@angular/core';
import * as import2 from '@angular/common';
import * as import3 from '@angular/material/input/input-container';
const styles_MdInputContainer:any[] = [import0.styles];
export const RenderType_MdInputContainer:import1.RendererType2 = import1.ɵcrt({
  encapsulation: 2,
  styles: styles_MdInputContainer,
  data: {}
}
);
function View_MdInputContainer_2(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
      (l()(),import1.ɵeld(0,(null as any),(null as any),1,'span',[[
        'class',
        'mat-placeholder-required'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import1.ɵted((null as any),['*']))
  ]
  ,(null as any),(null as any));
}
function View_MdInputContainer_1(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
      (l()(),import1.ɵeld(0,(null as any),(null as any),4,'label',[[
        'class',
        'mat-input-placeholder'
      ]
    ],[
      [
        1,
        'for',
        0
      ]
      ,
      [
        2,
        'mat-empty',
        (null as any)
      ]
      ,
      [
        2,
        'mat-focused',
        (null as any)
      ]
      ,
      [
        2,
        'mat-float',
        (null as any)
      ]
      ,
      [
        2,
        'mat-accent',
        (null as any)
      ]
      ,
      [
        2,
        'mat-warn',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    import1.ɵncd((null as any),2),
    (l()(),import1.ɵted((null as any),[
      '',
      ' '
    ]
    )),
    (l()(),import1.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MdInputContainer_2)),
    import1.ɵdid(8192,(null as any),0,import2.NgIf,[
      import1.ViewContainerRef,
      import1.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_7:any = co._mdInputChild.required;
    ck(v,4,0,currVal_7);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co._mdInputChild.id;
    const currVal_1:any = (co._mdInputChild.empty && !co._shouldAlwaysFloat);
    const currVal_2:any = co._mdInputChild.focused;
    const currVal_3:any = co._canPlaceholderFloat;
    const currVal_4:any = (co.dividerColor == 'accent');
    const currVal_5:any = (co.dividerColor == 'warn');
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5);
    const currVal_6:any = co._mdInputChild.placeholder;
    ck(v,2,0,currVal_6);
  });
}
function View_MdInputContainer_3(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
      (l()(),import1.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'mat-hint'
      ]
      ],[[
        1,
        'id',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    (l()(),import1.ɵted((null as any),[
      '',
      ''
    ]
    ))
  ]
  ,(null as any),(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co._hintLabelId;
    ck(v,0,0,currVal_0);
    const currVal_1:any = co.hintLabel;
    ck(v,1,0,currVal_1);
  });
}
export function View_MdInputContainer_0(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
      (l()(),import1.ɵeld(0,(null as any),(null as any),15,'div',[[
        'class',
        'mat-input-wrapper'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import1.ɵeld(0,(null as any),(null as any),9,'div',[[
        'class',
        'mat-input-table'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
      (l()(),import1.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'mat-input-prefix'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import1.ɵncd((null as any),0),
      (l()(),import1.ɵeld(0,(null as any),(null as any),4,'div',[[
        'class',
        'mat-input-infix'
      ]
      ],[[
        2,
        'mat-end',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import1.ɵncd((null as any),1),
      (l()(),import1.ɵeld(0,(null as any),(null as any),2,'span',[[
        'class',
        'mat-input-placeholder-wrapper'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import1.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MdInputContainer_1)),
    import1.ɵdid(8192,(null as any),0,import2.NgIf,[
      import1.ViewContainerRef,
      import1.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
      (l()(),import1.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'mat-input-suffix'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import1.ɵncd((null as any),3),
      (l()(),import1.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'mat-input-underline'
      ]
      ],[[
        2,
        'mat-disabled',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
      (l()(),import1.ɵeld(0,(null as any),(null as any),0,'span',[[
        'class',
        'mat-input-ripple'
      ]
    ],[
      [
        2,
        'mat-focused',
        (null as any)
      ]
      ,
      [
        2,
        'mat-accent',
        (null as any)
      ]
      ,
      [
        2,
        'mat-warn',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    (l()(),import1.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MdInputContainer_3)),
    import1.ɵdid(8192,(null as any),0,import2.NgIf,[
      import1.ViewContainerRef,
      import1.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any)),
    import1.ɵncd((null as any),4)
  ]
  ,(ck,v) => {
    var co:import3.MdInputContainer = v.component;
    const currVal_1:any = co._hasPlaceholder();
    ck(v,8,0,currVal_1);
    const currVal_6:any = (co.hintLabel != '');
    ck(v,14,0,currVal_6);
  },(ck,v) => {
    var co:import3.MdInputContainer = v.component;
    const currVal_0:any = (co.align == 'end');
    ck(v,4,0,currVal_0);
    const currVal_2:any = co._mdInputChild.disabled;
    ck(v,11,0,currVal_2);
    const currVal_3:any = co._mdInputChild.focused;
    const currVal_4:any = (co.dividerColor == 'accent');
    const currVal_5:any = (co.dividerColor == 'warn');
    ck(v,12,0,currVal_3,currVal_4,currVal_5);
  });
}
function View_MdInputContainer_Host_0(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
    (l()(),import1.ɵeld(0,(null as any),(null as any),4,'md-input-container',([] as any[]),[
      [
        2,
        'mat-input-container',
        (null as any)
      ]
      ,
      [
        1,
        'align',
        0
      ]
      ,
      [
        2,
        'ng-untouched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-touched',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pristine',
        (null as any)
      ]
      ,
      [
        2,
        'ng-dirty',
        (null as any)
      ]
      ,
      [
        2,
        'ng-valid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-invalid',
        (null as any)
      ]
      ,
      [
        2,
        'ng-pending',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import1.ɵnov(v,1)._focusInput()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },View_MdInputContainer_0,RenderType_MdInputContainer)),
    import1.ɵdid(548864,(null as any),3,import3.MdInputContainer,([] as any[]),(null as any),(null as any)),
    import1.ɵqud(167772160,1,{_mdInputChild: 0}),
    import1.ɵqud(167772160,2,{_placeholderChild: 0}),
    import1.ɵqud(301989888,3,{_hintChildren: 1})
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = true;
    const currVal_1:any = (null as any);
    const currVal_2:any = import1.ɵnov(v,1)._shouldForward('untouched');
    const currVal_3:any = import1.ɵnov(v,1)._shouldForward('touched');
    const currVal_4:any = import1.ɵnov(v,1)._shouldForward('pristine');
    const currVal_5:any = import1.ɵnov(v,1)._shouldForward('dirty');
    const currVal_6:any = import1.ɵnov(v,1)._shouldForward('valid');
    const currVal_7:any = import1.ɵnov(v,1)._shouldForward('invalid');
    const currVal_8:any = import1.ɵnov(v,1)._shouldForward('pending');
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7,currVal_8);
  });
}
export const MdInputContainerNgFactory:import1.ComponentFactory<import3.MdInputContainer> = import1.ɵccf('md-input-container, mat-input-container',import3.MdInputContainer,View_MdInputContainer_Host_0,{
  align: 'align',
  dividerColor: 'dividerColor',
  hintLabel: 'hintLabel',
  floatPlaceholder: 'floatPlaceholder'
}
,{},[
  '[mdPrefix], [matPrefix], [md-prefix]',
  '*',
  'md-placeholder, mat-placeholder',
  '[mdSuffix], [matSuffix], [md-suffix]',
  'md-hint, mat-hint'
]
);
