// @angular/material/input/input-container.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
import * as import0 from './input-container.css.ngstyle';
import * as import1 from '@angular/core';
import * as import2 from '@angular/common';
import * as import3 from '@angular/material/input/input-container';
var styles_MdInputContainer = [import0.styles];
export var RenderType_MdInputContainer = import1.ɵcrt({
    encapsulation: 2,
    styles: styles_MdInputContainer,
    data: {}
});
function View_MdInputContainer_2(l) {
    return import1.ɵvid(0, [
        (l()(), import1.ɵeld(0, null, null, 1, 'span', [[
                'class',
                'mat-placeholder-required'
            ]
        ], null, null, null, null, null)),
        (l()(), import1.ɵted(null, ['*']))
    ], null, null);
}
function View_MdInputContainer_1(l) {
    return import1.ɵvid(0, [
        (l()(), import1.ɵeld(0, null, null, 4, 'label', [[
                'class',
                'mat-input-placeholder'
            ]
        ], [
            [
                1,
                'for',
                0
            ],
            [
                2,
                'mat-empty',
                null
            ],
            [
                2,
                'mat-focused',
                null
            ],
            [
                2,
                'mat-float',
                null
            ],
            [
                2,
                'mat-accent',
                null
            ],
            [
                2,
                'mat-warn',
                null
            ]
        ], null, null, null, null)),
        import1.ɵncd(null, 2),
        (l()(), import1.ɵted(null, [
            '',
            ' '
        ])),
        (l()(), import1.ɵand(8388608, null, null, 1, null, View_MdInputContainer_2)),
        import1.ɵdid(8192, null, 0, import2.NgIf, [
            import1.ViewContainerRef,
            import1.TemplateRef
        ], { ngIf: [
                0,
                'ngIf'
            ]
        }, null)
    ], function (ck, v) {
        var co = v.component;
        var currVal_7 = co._mdInputChild.required;
        ck(v, 4, 0, currVal_7);
    }, function (ck, v) {
        var co = v.component;
        var currVal_0 = co._mdInputChild.id;
        var currVal_1 = (co._mdInputChild.empty && !co._shouldAlwaysFloat);
        var currVal_2 = co._mdInputChild.focused;
        var currVal_3 = co._canPlaceholderFloat;
        var currVal_4 = (co.dividerColor == 'accent');
        var currVal_5 = (co.dividerColor == 'warn');
        ck(v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4, currVal_5);
        var currVal_6 = co._mdInputChild.placeholder;
        ck(v, 2, 0, currVal_6);
    });
}
function View_MdInputContainer_3(l) {
    return import1.ɵvid(0, [
        (l()(), import1.ɵeld(0, null, null, 1, 'div', [[
                'class',
                'mat-hint'
            ]
        ], [[
                1,
                'id',
                0
            ]
        ], null, null, null, null)),
        (l()(), import1.ɵted(null, [
            '',
            ''
        ]))
    ], null, function (ck, v) {
        var co = v.component;
        var currVal_0 = co._hintLabelId;
        ck(v, 0, 0, currVal_0);
        var currVal_1 = co.hintLabel;
        ck(v, 1, 0, currVal_1);
    });
}
export function View_MdInputContainer_0(l) {
    return import1.ɵvid(0, [
        (l()(), import1.ɵeld(0, null, null, 15, 'div', [[
                'class',
                'mat-input-wrapper'
            ]
        ], null, null, null, null, null)),
        (l()(), import1.ɵeld(0, null, null, 9, 'div', [[
                'class',
                'mat-input-table'
            ]
        ], null, null, null, null, null)),
        (l()(), import1.ɵeld(0, null, null, 1, 'div', [[
                'class',
                'mat-input-prefix'
            ]
        ], null, null, null, null, null)),
        import1.ɵncd(null, 0),
        (l()(), import1.ɵeld(0, null, null, 4, 'div', [[
                'class',
                'mat-input-infix'
            ]
        ], [[
                2,
                'mat-end',
                null
            ]
        ], null, null, null, null)),
        import1.ɵncd(null, 1),
        (l()(), import1.ɵeld(0, null, null, 2, 'span', [[
                'class',
                'mat-input-placeholder-wrapper'
            ]
        ], null, null, null, null, null)),
        (l()(), import1.ɵand(8388608, null, null, 1, null, View_MdInputContainer_1)),
        import1.ɵdid(8192, null, 0, import2.NgIf, [
            import1.ViewContainerRef,
            import1.TemplateRef
        ], { ngIf: [
                0,
                'ngIf'
            ]
        }, null),
        (l()(), import1.ɵeld(0, null, null, 1, 'div', [[
                'class',
                'mat-input-suffix'
            ]
        ], null, null, null, null, null)),
        import1.ɵncd(null, 3),
        (l()(), import1.ɵeld(0, null, null, 1, 'div', [[
                'class',
                'mat-input-underline'
            ]
        ], [[
                2,
                'mat-disabled',
                null
            ]
        ], null, null, null, null)),
        (l()(), import1.ɵeld(0, null, null, 0, 'span', [[
                'class',
                'mat-input-ripple'
            ]
        ], [
            [
                2,
                'mat-focused',
                null
            ],
            [
                2,
                'mat-accent',
                null
            ],
            [
                2,
                'mat-warn',
                null
            ]
        ], null, null, null, null)),
        (l()(), import1.ɵand(8388608, null, null, 1, null, View_MdInputContainer_3)),
        import1.ɵdid(8192, null, 0, import2.NgIf, [
            import1.ViewContainerRef,
            import1.TemplateRef
        ], { ngIf: [
                0,
                'ngIf'
            ]
        }, null),
        import1.ɵncd(null, 4)
    ], function (ck, v) {
        var co = v.component;
        var currVal_1 = co._hasPlaceholder();
        ck(v, 8, 0, currVal_1);
        var currVal_6 = (co.hintLabel != '');
        ck(v, 14, 0, currVal_6);
    }, function (ck, v) {
        var co = v.component;
        var currVal_0 = (co.align == 'end');
        ck(v, 4, 0, currVal_0);
        var currVal_2 = co._mdInputChild.disabled;
        ck(v, 11, 0, currVal_2);
        var currVal_3 = co._mdInputChild.focused;
        var currVal_4 = (co.dividerColor == 'accent');
        var currVal_5 = (co.dividerColor == 'warn');
        ck(v, 12, 0, currVal_3, currVal_4, currVal_5);
    });
}
function View_MdInputContainer_Host_0(l) {
    return import1.ɵvid(0, [
        (l()(), import1.ɵeld(0, null, null, 4, 'md-input-container', [], [
            [
                2,
                'mat-input-container',
                null
            ],
            [
                1,
                'align',
                0
            ],
            [
                2,
                'ng-untouched',
                null
            ],
            [
                2,
                'ng-touched',
                null
            ],
            [
                2,
                'ng-pristine',
                null
            ],
            [
                2,
                'ng-dirty',
                null
            ],
            [
                2,
                'ng-valid',
                null
            ],
            [
                2,
                'ng-invalid',
                null
            ],
            [
                2,
                'ng-pending',
                null
            ]
        ], [[
                null,
                'click'
            ]
        ], function (v, en, $event) {
            var ad = true;
            if (('click' === en)) {
                var pd_0 = (import1.ɵnov(v, 1)._focusInput() !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, View_MdInputContainer_0, RenderType_MdInputContainer)),
        import1.ɵdid(548864, null, 3, import3.MdInputContainer, [], null, null),
        import1.ɵqud(167772160, 1, { _mdInputChild: 0 }),
        import1.ɵqud(167772160, 2, { _placeholderChild: 0 }),
        import1.ɵqud(301989888, 3, { _hintChildren: 1 })
    ], null, function (ck, v) {
        var currVal_0 = true;
        var currVal_1 = null;
        var currVal_2 = import1.ɵnov(v, 1)._shouldForward('untouched');
        var currVal_3 = import1.ɵnov(v, 1)._shouldForward('touched');
        var currVal_4 = import1.ɵnov(v, 1)._shouldForward('pristine');
        var currVal_5 = import1.ɵnov(v, 1)._shouldForward('dirty');
        var currVal_6 = import1.ɵnov(v, 1)._shouldForward('valid');
        var currVal_7 = import1.ɵnov(v, 1)._shouldForward('invalid');
        var currVal_8 = import1.ɵnov(v, 1)._shouldForward('pending');
        ck(v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4, currVal_5, currVal_6, currVal_7, currVal_8);
    });
}
export var MdInputContainerNgFactory = import1.ɵccf('md-input-container, mat-input-container', import3.MdInputContainer, View_MdInputContainer_Host_0, {
    align: 'align',
    dividerColor: 'dividerColor',
    hintLabel: 'hintLabel',
    floatPlaceholder: 'floatPlaceholder'
}, {}, [
    '[mdPrefix], [matPrefix], [md-prefix]',
    '*',
    'md-placeholder, mat-placeholder',
    '[mdSuffix], [matSuffix], [md-suffix]',
    'md-hint, mat-hint'
]);
