// @angular/material/progress-spinner/progress-spinner.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/material/progress-spinner/progress-spinner';
import * as import2 from '@angular/material/core/compatibility/compatibility';
import * as import3 from './progress-spinner.css.shim.ngstyle';
class MdProgressSpinnerModuleInjector extends import0.ɵNgModuleInjector<import1.MdProgressSpinnerModule> {
  _CompatibilityModule_0:import2.CompatibilityModule;
  _MdProgressSpinnerModule_1:import1.MdProgressSpinnerModule;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  createInternal():import1.MdProgressSpinnerModule {
    this._CompatibilityModule_0 = new import2.CompatibilityModule();
    this._MdProgressSpinnerModule_1 = new import1.MdProgressSpinnerModule();
    return this._MdProgressSpinnerModule_1;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CompatibilityModule)) { return this._CompatibilityModule_0; }
    if ((token === import1.MdProgressSpinnerModule)) { return this._MdProgressSpinnerModule_1; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const MdProgressSpinnerModuleNgFactory:import0.NgModuleFactory<import1.MdProgressSpinnerModule> = new import0.NgModuleFactory<any>(MdProgressSpinnerModuleInjector,import1.MdProgressSpinnerModule);
const styles_MdProgressSpinner:any[] = [import3.styles];
export const RenderType_MdProgressSpinner:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 0,
  styles: styles_MdProgressSpinner,
  data: {}
}
);
export function View_MdProgressSpinner_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(2,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,':svg:svg',[
      [
        'preserveAspectRatio',
        'xMidYMid meet'
      ]
      ,
      [
        'viewBox',
        '0 0 100 100'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,':svg:path',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any)))
  ]
  ,(null as any),(null as any));
}
function View_MdProgressSpinner_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'md-progress-spinner',[[
        'role',
        'progressbar'
      ]
    ],[
      [
        1,
        'aria-valuemin',
        0
      ]
      ,
      [
        1,
        'aria-valuemax',
        0
      ]
      ,
      [
        1,
        'aria-valuenow',
        0
      ]
      ,
      [
        1,
        'mode',
        0
      ]

    ]
    ,(null as any),(null as any),View_MdProgressSpinner_0,RenderType_MdProgressSpinner)),
    import0.ɵdid(90112,(null as any),0,import1.MdProgressSpinner,[
      import0.NgZone,
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1)._ariaValueMin;
    const currVal_1:any = import0.ɵnov(v,1)._ariaValueMax;
    const currVal_2:any = import0.ɵnov(v,1).value;
    const currVal_3:any = import0.ɵnov(v,1).mode;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3);
  });
}
export const MdProgressSpinnerNgFactory:import0.ComponentFactory<import1.MdProgressSpinner> = import0.ɵccf('md-progress-spinner, mat-progress-spinner, md-progress-circle, mat-progress-circle',import1.MdProgressSpinner,View_MdProgressSpinner_Host_0,{
  color: 'color',
  value: 'value',
  mode: 'mode'
}
,{},([] as any[]));
const styles_MdSpinner:any[] = [import3.styles];
export const RenderType_MdSpinner:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 0,
  styles: styles_MdSpinner,
  data: {}
}
);
export function View_MdSpinner_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,':svg:svg',[
      [
        'preserveAspectRatio',
        'xMidYMid meet'
      ]
      ,
      [
        'viewBox',
        '0 0 100 100'
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import0.ɵeld(0,(null as any),(null as any),0,':svg:path',([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any)))
  ]
  ,(null as any),(null as any));
}
function View_MdSpinner_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'md-spinner',[
      [
        'mode',
        'indeterminate'
      ]
      ,
      [
        'role',
        'progressbar'
      ]

    ]
    ,[
      [
        2,
        'mat-spinner',
        (null as any)
      ]
      ,
      [
        1,
        'aria-valuenow',
        0
      ]
      ,
      [
        1,
        'mode',
        0
      ]

    ]
    ,(null as any),(null as any),View_MdSpinner_0,RenderType_MdSpinner)),
    import0.ɵdid(90112,(null as any),0,import1.MdSpinner,[
      import0.ElementRef,
      import0.NgZone,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = true;
    const currVal_1:any = import0.ɵnov(v,1).value;
    const currVal_2:any = import0.ɵnov(v,1).mode;
    ck(v,0,0,currVal_0,currVal_1,currVal_2);
  });
}
export const MdSpinnerNgFactory:import0.ComponentFactory<import1.MdSpinner> = import0.ɵccf('md-spinner, mat-spinner',import1.MdSpinner,View_MdSpinner_Host_0,{
  color: 'color',
  value: 'value',
  mode: 'mode'
}
,{},([] as any[]));
