// @angular/material/tabs/tab-header.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from './tab-header.css.ngstyle';
import * as import1 from '@angular/core';
import * as import2 from '@angular/material/tabs/tab-header';
import * as import3 from '@angular/material/core/ripple/ripple';
import * as import4 from '@angular/material/core/overlay/position/viewport-ruler';
import * as import5 from '@angular/material/core/observe-content/observe-content';
import * as import6 from '@angular/material/core/compatibility/compatibility';
import * as import7 from '@angular/material/tabs/ink-bar';
import * as import8 from '@angular/material/core/rtl/dir';
const styles_MdTabHeader:any[] = [import0.styles];
export const RenderType_MdTabHeader:import1.RendererType2 = import1.ɵcrt({
  encapsulation: 2,
  styles: styles_MdTabHeader,
  data: {}
}
);
export function View_MdTabHeader_0(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
    import1.ɵqud(201326592,1,{_inkBar: 0}),
    import1.ɵqud(201326592,2,{_tabListContainer: 0}),
    import1.ɵqud(201326592,3,{_tabList: 0}),
    (l()(),import1.ɵeld(0,(null as any),(null as any),2,'div',[
      [
        'aria-hidden',
        'true'
      ]
      ,
      [
        'class',
        'mat-tab-header-pagination mat-tab-header-pagination-before mat-elevation-z4'
      ]
      ,
      [
        'md-ripple',
        ''
      ]

    ]
    ,[
      [
        2,
        'mat-tab-header-pagination-disabled',
        (null as any)
      ]
      ,
      [
        2,
        'mat-ripple',
        (null as any)
      ]
      ,
      [
        2,
        'mat-ripple-unbounded',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import2.MdTabHeader = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co._scrollHeader('before')) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import1.ɵdid(335872,(null as any),0,import3.MdRipple,[
      import1.ElementRef,
      import1.NgZone,
      import4.ViewportRuler
    ]
      ,{disabled: [
        0,
        'disabled'
      ]
    },(null as any)),
      (l()(),import1.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'mat-tab-header-pagination-chevron'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    (l()(),import1.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'tabListContainer',
        1
      ]

    ]
      ,(null as any),7,'div',[[
        'class',
        'mat-tab-label-container'
      ]
      ],(null as any),[[
        (null as any),
        'keydown'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import2.MdTabHeader = v.component;
      if (('keydown' === en)) {
        const pd_0:any = ((<any>co._handleKeydown($event)) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    (l()(),import1.ɵeld(0,[
      [
        3,
        0
      ]
      ,
      [
        'tabList',
        1
      ]

    ]
    ,(null as any),6,'div',[
      [
        'class',
        'mat-tab-list'
      ]
      ,
      [
        'role',
        'tablist'
      ]

    ]
      ,(null as any),[[
        (null as any),
        'cdkObserveContent'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import2.MdTabHeader = v.component;
      if (('cdkObserveContent' === en)) {
        const pd_0:any = ((<any>co._onContentChanges()) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import1.ɵdid(598016,(null as any),0,import5.ObserveContent,[import1.ElementRef],(null as any),{event: 'cdkObserveContent'}),
      (l()(),import1.ɵeld(0,(null as any),(null as any),1,'div',[[
        'class',
        'mat-tab-labels'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any))),
    import1.ɵncd((null as any),0),
      (l()(),import1.ɵeld(0,(null as any),(null as any),2,'md-ink-bar',([] as any[]),[[
        2,
        'mat-ink-bar',
        (null as any)
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
      import1.ɵdid(8192,(null as any),0,import6.MdPrefixRejector,[[
        2,
        import6.MATERIAL_COMPATIBILITY_MODE
      ]
    ],(null as any),(null as any)),
      import1.ɵdid(8192,[[
        1,
        4
      ]
    ],0,import7.MdInkBar,[
      import1.Renderer,
      import1.ElementRef
    ]
    ,(null as any),(null as any)),
    (l()(),import1.ɵeld(0,(null as any),(null as any),2,'div',[
      [
        'aria-hidden',
        'true'
      ]
      ,
      [
        'class',
        'mat-tab-header-pagination mat-tab-header-pagination-after mat-elevation-z4'
      ]
      ,
      [
        'md-ripple',
        ''
      ]

    ]
    ,[
      [
        2,
        'mat-tab-header-pagination-disabled',
        (null as any)
      ]
      ,
      [
        2,
        'mat-ripple',
        (null as any)
      ]
      ,
      [
        2,
        'mat-ripple-unbounded',
        (null as any)
      ]

    ]
      ,[[
        (null as any),
        'click'
      ]
    ],(v,en,$event) => {
      var ad:boolean = true;
      var co:import2.MdTabHeader = v.component;
      if (('click' === en)) {
        const pd_0:any = ((<any>co._scrollHeader('after')) !== false);
        ad = (pd_0 && ad);
      }
      return ad;
    },(null as any),(null as any))),
    import1.ɵdid(335872,(null as any),0,import3.MdRipple,[
      import1.ElementRef,
      import1.NgZone,
      import4.ViewportRuler
    ]
      ,{disabled: [
        0,
        'disabled'
      ]
    },(null as any)),
      (l()(),import1.ɵeld(0,(null as any),(null as any),0,'div',[[
        'class',
        'mat-tab-header-pagination-chevron'
      ]
    ],(null as any),(null as any),(null as any),(null as any),(null as any)))
  ]
  ,(ck,v) => {
    var co:import2.MdTabHeader = v.component;
    const currVal_3:any = co._disableScrollBefore;
    ck(v,4,0,currVal_3);
    const currVal_8:any = co._disableScrollAfter;
    ck(v,15,0,currVal_8);
  },(ck,v) => {
    var co:import2.MdTabHeader = v.component;
    const currVal_0:any = co._disableScrollBefore;
    const currVal_1:any = true;
    const currVal_2:any = import1.ɵnov(v,4).unbounded;
    ck(v,3,0,currVal_0,currVal_1,currVal_2);
    const currVal_4:any = true;
    ck(v,11,0,currVal_4);
    const currVal_5:any = co._disableScrollAfter;
    const currVal_6:any = true;
    const currVal_7:any = import1.ɵnov(v,15).unbounded;
    ck(v,14,0,currVal_5,currVal_6,currVal_7);
  });
}
function View_MdTabHeader_Host_0(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
      (l()(),import1.ɵeld(0,(null as any),(null as any),2,'md-tab-header',[[
        'class',
        'mat-tab-header'
      ]
    ],[
      [
        2,
        'mat-tab-header-pagination-controls-enabled',
        (null as any)
      ]
      ,
      [
        2,
        'mat-tab-header-rtl',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),View_MdTabHeader_0,RenderType_MdTabHeader)),
    import1.ɵdid(1597440,(null as any),1,import2.MdTabHeader,[
      import1.NgZone,
      import1.ElementRef,
      [
        2,
        import8.Dir
      ]

    ]
    ,(null as any),(null as any)),
    import1.ɵqud(301989888,1,{_labelWrappers: 1})
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import1.ɵnov(v,1)._showPaginationControls;
    const currVal_1:any = (import1.ɵnov(v,1)._getLayoutDirection() == 'rtl');
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export const MdTabHeaderNgFactory:import1.ComponentFactory<import2.MdTabHeader> = import1.ɵccf('md-tab-header, mat-tab-header',import2.MdTabHeader,View_MdTabHeader_Host_0,{selectedIndex: 'selectedIndex'},{
  selectFocusedIndex: 'selectFocusedIndex',
  indexFocused: 'indexFocused'
}
,['*']);
