// @angular/material/tabs/tab-group.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import * as import0 from '@angular/core';
import * as import1 from '@angular/material/tabs/tab-group';
import * as import2 from '@angular/common';
import * as import3 from '@angular/material/core/portal/portal-directives';
import * as import4 from '@angular/material/core/compatibility/compatibility';
import * as import5 from '@angular/material/core/ripple/ripple';
import * as import6 from '@angular/material/core/observe-content/observe-content';
import * as import7 from '@angular/material/core/overlay/scroll/scroll-dispatcher';
import * as import8 from '@angular/material/core/overlay/position/viewport-ruler';
import * as import9 from './tab-group.css.shim.ngstyle';
import * as import10 from '@angular/material/tabs/tab-label-wrapper';
import * as import11 from './tab-body.ngfactory';
import * as import12 from '@angular/material/tabs/tab-body';
import * as import13 from '@angular/material/core/rtl/dir';
import * as import14 from './tab-header.ngfactory';
import * as import15 from '@angular/material/tabs/tab-header';
var MdTabsModuleInjector = (function (_super) {
    __extends(MdTabsModuleInjector, _super);
    function MdTabsModuleInjector(parent) {
        return _super.call(this, parent, [], []) || this;
    }
    Object.defineProperty(MdTabsModuleInjector.prototype, "_NgLocalization_6", {
        get: function () {
            if ((this.__NgLocalization_6 == null)) {
                (this.__NgLocalization_6 = new import2.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID)));
            }
            return this.__NgLocalization_6;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(MdTabsModuleInjector.prototype, "_ScrollDispatcher_7", {
        get: function () {
            if ((this.__ScrollDispatcher_7 == null)) {
                (this.__ScrollDispatcher_7 = import7.SCROLL_DISPATCHER_PROVIDER_FACTORY(this.parent.get(import7.ScrollDispatcher, null)));
            }
            return this.__ScrollDispatcher_7;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(MdTabsModuleInjector.prototype, "_ViewportRuler_8", {
        get: function () {
            if ((this.__ViewportRuler_8 == null)) {
                (this.__ViewportRuler_8 = import8.VIEWPORT_RULER_PROVIDER_FACTORY(this.parent.get(import8.ViewportRuler, null), this._ScrollDispatcher_7));
            }
            return this.__ViewportRuler_8;
        },
        enumerable: true,
        configurable: true
    });
    MdTabsModuleInjector.prototype.createInternal = function () {
        this._CommonModule_0 = new import2.CommonModule();
        this._PortalModule_1 = new import3.PortalModule();
        this._CompatibilityModule_2 = new import4.CompatibilityModule();
        this._MdRippleModule_3 = new import5.MdRippleModule();
        this._ObserveContentModule_4 = new import6.ObserveContentModule();
        this._MdTabsModule_5 = new import1.MdTabsModule();
        return this._MdTabsModule_5;
    };
    MdTabsModuleInjector.prototype.getInternal = function (token, notFoundResult) {
        if ((token === import2.CommonModule)) {
            return this._CommonModule_0;
        }
        if ((token === import3.PortalModule)) {
            return this._PortalModule_1;
        }
        if ((token === import4.CompatibilityModule)) {
            return this._CompatibilityModule_2;
        }
        if ((token === import5.MdRippleModule)) {
            return this._MdRippleModule_3;
        }
        if ((token === import6.ObserveContentModule)) {
            return this._ObserveContentModule_4;
        }
        if ((token === import1.MdTabsModule)) {
            return this._MdTabsModule_5;
        }
        if ((token === import2.NgLocalization)) {
            return this._NgLocalization_6;
        }
        if ((token === import7.ScrollDispatcher)) {
            return this._ScrollDispatcher_7;
        }
        if ((token === import8.ViewportRuler)) {
            return this._ViewportRuler_8;
        }
        return notFoundResult;
    };
    MdTabsModuleInjector.prototype.destroyInternal = function () {
    };
    return MdTabsModuleInjector;
}(import0.ɵNgModuleInjector));
export var MdTabsModuleNgFactory = new import0.NgModuleFactory(MdTabsModuleInjector, import1.MdTabsModule);
var styles_MdTabGroup = [import9.styles];
export var RenderType_MdTabGroup = import0.ɵcrt({
    encapsulation: 0,
    styles: styles_MdTabGroup,
    data: {}
});
function View_MdTabGroup_3(l) {
    return import0.ɵvid(0, [(l()(), import0.ɵand(0, null, null, 0))], null, null);
}
function View_MdTabGroup_2(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_MdTabGroup_3)),
        import0.ɵdid(73728, null, 0, import3.PortalHostDirective, [
            import0.ComponentFactoryResolver,
            import0.ViewContainerRef
        ], { portal: [
                0,
                'portal'
            ]
        }, null),
        (l()(), import0.ɵand(0, null, null, 0))
    ], function (ck, v) {
        var currVal_0 = v.parent.context.$implicit.templateLabel;
        ck(v, 1, 0, currVal_0);
    }, null);
}
function View_MdTabGroup_4(l) {
    return import0.ɵvid(0, [(l()(), import0.ɵted(null, [
            '',
            ''
        ]))], null, function (ck, v) {
        var currVal_0 = v.parent.context.$implicit.textLabel;
        ck(v, 0, 0, currVal_0);
    });
}
function View_MdTabGroup_1(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 6, 'div', [
            [
                'class',
                'mat-tab-label'
            ],
            [
                'md-ripple',
                ''
            ],
            [
                'md-tab-label-wrapper',
                ''
            ],
            [
                'role',
                'tab'
            ]
        ], [
            [
                8,
                'id',
                0
            ],
            [
                8,
                'tabIndex',
                0
            ],
            [
                1,
                'aria-controls',
                0
            ],
            [
                1,
                'aria-selected',
                0
            ],
            [
                2,
                'mat-tab-label-active',
                null
            ],
            [
                2,
                'mat-ripple',
                null
            ],
            [
                2,
                'mat-ripple-unbounded',
                null
            ],
            [
                2,
                'mat-tab-disabled',
                null
            ]
        ], [[
                null,
                'click'
            ]
        ], function (v, en, $event) {
            var ad = true;
            var co = v.component;
            if (('click' === en)) {
                var pd_0 = ((import0.ɵnov(v.parent, 3).focusIndex = (co.selectedIndex = v.context.index)) !== false);
                ad = (pd_0 && ad);
            }
            return ad;
        }, null, null)),
        import0.ɵdid(335872, null, 0, import5.MdRipple, [
            import0.ElementRef,
            import0.NgZone,
            import8.ViewportRuler
        ], null, null),
        import0.ɵdid(8192, [[
                2,
                4
            ]
        ], 0, import10.MdTabLabelWrapper, [
            import0.ElementRef,
            import0.Renderer
        ], { disabled: [
                0,
                'disabled'
            ]
        }, null),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_MdTabGroup_2)),
        import0.ɵdid(8192, null, 0, import2.NgIf, [
            import0.ViewContainerRef,
            import0.TemplateRef
        ], { ngIf: [
                0,
                'ngIf'
            ]
        }, null),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_MdTabGroup_4)),
        import0.ɵdid(8192, null, 0, import2.NgIf, [
            import0.ViewContainerRef,
            import0.TemplateRef
        ], { ngIf: [
                0,
                'ngIf'
            ]
        }, null)
    ], function (ck, v) {
        var currVal_8 = v.context.$implicit.disabled;
        ck(v, 2, 0, currVal_8);
        var currVal_9 = v.context.$implicit.templateLabel;
        ck(v, 4, 0, currVal_9);
        var currVal_10 = !v.context.$implicit.templateLabel;
        ck(v, 6, 0, currVal_10);
    }, function (ck, v) {
        var co = v.component;
        var currVal_0 = co._getTabLabelId(v.context.index);
        var currVal_1 = ((co.selectedIndex == v.context.index) ? 0 : (0 - 1));
        var currVal_2 = co._getTabContentId(v.context.index);
        var currVal_3 = (co.selectedIndex == v.context.index);
        var currVal_4 = (co.selectedIndex == v.context.index);
        var currVal_5 = true;
        var currVal_6 = import0.ɵnov(v, 1).unbounded;
        var currVal_7 = import0.ɵnov(v, 2).disabled;
        ck(v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3, currVal_4, currVal_5, currVal_6, currVal_7);
    });
}
function View_MdTabGroup_5(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 2, 'md-tab-body', [[
                'role',
                'tabpanel'
            ]
        ], [
            [
                8,
                'id',
                0
            ],
            [
                1,
                'aria-labelledby',
                0
            ],
            [
                2,
                'mat-tab-body-active',
                null
            ],
            [
                2,
                'mat-tab-body',
                null
            ]
        ], [
            [
                null,
                'onCentered'
            ],
            [
                null,
                'onCentering'
            ]
        ], function (v, en, $event) {
            var ad = true;
            var co = v.component;
            if (('onCentered' === en)) {
                var pd_0 = (co._removeTabBodyWrapperHeight() !== false);
                ad = (pd_0 && ad);
            }
            if (('onCentering' === en)) {
                var pd_1 = (co._setTabBodyWrapperHeight($event) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, import11.View_MdTabBody_0, import11.RenderType_MdTabBody)),
        import0.ɵdid(8192, null, 0, import4.MdPrefixRejector, [[
                2,
                import4.MATERIAL_COMPATIBILITY_MODE
            ]
        ], null, null),
        import0.ɵdid(5300224, null, 0, import12.MdTabBody, [
            [
                2,
                import13.Dir
            ],
            import0.ElementRef,
            import0.ChangeDetectorRef
        ], {
            _content: [
                0,
                '_content'
            ],
            position: [
                1,
                'position'
            ],
            origin: [
                2,
                'origin'
            ]
        }, {
            onCentering: 'onCentering',
            onCentered: 'onCentered'
        })
    ], function (ck, v) {
        var currVal_4 = v.context.$implicit.content;
        var currVal_5 = v.context.$implicit.position;
        var currVal_6 = v.context.$implicit.origin;
        ck(v, 2, 0, currVal_4, currVal_5, currVal_6);
    }, function (ck, v) {
        var co = v.component;
        var currVal_0 = co._getTabContentId(v.context.index);
        var currVal_1 = co._getTabLabelId(v.context.index);
        var currVal_2 = (co.selectedIndex == v.context.index);
        var currVal_3 = true;
        ck(v, 0, 0, currVal_0, currVal_1, currVal_2, currVal_3);
    });
}
export function View_MdTabGroup_0(l) {
    return import0.ɵvid(0, [
        import0.ɵqud(201326592, 1, { _tabBodyWrapper: 0 }),
        (l()(), import0.ɵeld(0, null, null, 5, 'md-tab-header', [[
                'class',
                'mat-tab-header'
            ]
        ], [
            [
                2,
                'mat-tab-header-pagination-controls-enabled',
                null
            ],
            [
                2,
                'mat-tab-header-rtl',
                null
            ]
        ], [
            [
                null,
                'indexFocused'
            ],
            [
                null,
                'selectFocusedIndex'
            ]
        ], function (v, en, $event) {
            var ad = true;
            var co = v.component;
            if (('indexFocused' === en)) {
                var pd_0 = (co._focusChanged($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('selectFocusedIndex' === en)) {
                var pd_1 = ((co.selectedIndex = $event) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, import14.View_MdTabHeader_0, import14.RenderType_MdTabHeader)),
        import0.ɵdid(8192, null, 0, import4.MdPrefixRejector, [[
                2,
                import4.MATERIAL_COMPATIBILITY_MODE
            ]
        ], null, null),
        import0.ɵdid(1597440, [[
                'tabHeader',
                4
            ]
        ], 1, import15.MdTabHeader, [
            import0.NgZone,
            import0.ElementRef,
            [
                2,
                import13.Dir
            ]
        ], { selectedIndex: [
                0,
                'selectedIndex'
            ]
        }, {
            selectFocusedIndex: 'selectFocusedIndex',
            indexFocused: 'indexFocused'
        }),
        import0.ɵqud(301989888, 2, { _labelWrappers: 1 }),
        (l()(), import0.ɵand(8388608, null, 0, 1, null, View_MdTabGroup_1)),
        import0.ɵdid(401408, null, 0, import2.NgForOf, [
            import0.ViewContainerRef,
            import0.TemplateRef,
            import0.IterableDiffers
        ], { ngForOf: [
                0,
                'ngForOf'
            ]
        }, null),
        (l()(), import0.ɵeld(0, [
            [
                1,
                0
            ],
            [
                'tabBodyWrapper',
                1
            ]
        ], null, 2, 'div', [[
                'class',
                'mat-tab-body-wrapper'
            ]
        ], null, null, null, null, null)),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_MdTabGroup_5)),
        import0.ɵdid(401408, null, 0, import2.NgForOf, [
            import0.ViewContainerRef,
            import0.TemplateRef,
            import0.IterableDiffers
        ], { ngForOf: [
                0,
                'ngForOf'
            ]
        }, null)
    ], function (ck, v) {
        var co = v.component;
        var currVal_2 = co.selectedIndex;
        ck(v, 3, 0, currVal_2);
        var currVal_3 = co._tabs;
        ck(v, 6, 0, currVal_3);
        var currVal_4 = co._tabs;
        ck(v, 9, 0, currVal_4);
    }, function (ck, v) {
        var currVal_0 = import0.ɵnov(v, 3)._showPaginationControls;
        var currVal_1 = (import0.ɵnov(v, 3)._getLayoutDirection() == 'rtl');
        ck(v, 1, 0, currVal_0, currVal_1);
    });
}
function View_MdTabGroup_Host_0(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 2, 'md-tab-group', [], [
            [
                2,
                'mat-tab-group-dynamic-height',
                null
            ],
            [
                2,
                'mat-tab-group-inverted-header',
                null
            ]
        ], null, null, View_MdTabGroup_0, RenderType_MdTabGroup)),
        import0.ɵdid(5267456, null, 1, import1.MdTabGroup, [import0.Renderer], null, null),
        import0.ɵqud(301989888, 1, { _tabs: 1 })
    ], null, function (ck, v) {
        var currVal_0 = import0.ɵnov(v, 1).dynamicHeight;
        var currVal_1 = (import0.ɵnov(v, 1).headerPosition === 'below');
        ck(v, 0, 0, currVal_0, currVal_1);
    });
}
export var MdTabGroupNgFactory = import0.ɵccf('md-tab-group, mat-tab-group', import1.MdTabGroup, View_MdTabGroup_Host_0, {
    dynamicHeight: 'dynamicHeight',
    _dynamicHeightDeprecated: 'md-dynamic-height',
    selectedIndex: 'selectedIndex',
    headerPosition: 'headerPosition'
}, {
    selectedIndexChange: 'selectedIndexChange',
    focusChange: 'focusChange',
    selectChange: 'selectChange'
}, []);
