// @angular/material/tabs/tab-body.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
import * as import0 from '@angular/core';
import * as import1 from '@angular/material/tabs/tab-body';
import * as import2 from '@angular/material/core/portal/portal-directives';
import * as import3 from '@angular/material/core/rtl/dir';
var styles_MdTabBody = [];
export var RenderType_MdTabBody = import0.ɵcrt({
    encapsulation: 2,
    styles: styles_MdTabBody,
    data: { 'animation': [{
                name: 'translateTab',
                definitions: [
                    {
                        type: 0,
                        name: 'left',
                        styles: {
                            type: 6,
                            styles: { transform: 'translate3d(-100%, 0, 0)' }
                        }
                    },
                    {
                        type: 0,
                        name: 'left-origin-center',
                        styles: {
                            type: 6,
                            styles: { transform: 'translate3d(0, 0, 0)' }
                        }
                    },
                    {
                        type: 0,
                        name: 'right-origin-center',
                        styles: {
                            type: 6,
                            styles: { transform: 'translate3d(0, 0, 0)' }
                        }
                    },
                    {
                        type: 0,
                        name: 'center',
                        styles: {
                            type: 6,
                            styles: { transform: 'translate3d(0, 0, 0)' }
                        }
                    },
                    {
                        type: 0,
                        name: 'right',
                        styles: {
                            type: 6,
                            styles: { transform: 'translate3d(100%, 0, 0)' }
                        }
                    },
                    {
                        type: 1,
                        expr: '* => left, * => right, left => center, right => center',
                        animation: {
                            type: 4,
                            styles: null,
                            timings: '500ms cubic-bezier(0.35, 0, 0.25, 1)'
                        }
                    },
                    {
                        type: 1,
                        expr: 'void => left-origin-center',
                        animation: [
                            {
                                type: 6,
                                styles: { transform: 'translate3d(-100%, 0, 0)' }
                            },
                            {
                                type: 4,
                                styles: null,
                                timings: '500ms cubic-bezier(0.35, 0, 0.25, 1)'
                            }
                        ]
                    },
                    {
                        type: 1,
                        expr: 'void => right-origin-center',
                        animation: [
                            {
                                type: 6,
                                styles: { transform: 'translate3d(100%, 0, 0)' }
                            },
                            {
                                type: 4,
                                styles: null,
                                timings: '500ms cubic-bezier(0.35, 0, 0.25, 1)'
                            }
                        ]
                    }
                ]
            }
        ] }
});
function View_MdTabBody_1(l) {
    return import0.ɵvid(0, [(l()(), import0.ɵand(0, null, null, 0))], null, null);
}
export function View_MdTabBody_0(l) {
    return import0.ɵvid(0, [
        import0.ɵqud(201326592, 1, { _portalHost: 0 }),
        (l()(), import0.ɵeld(0, [[
                'content',
                1
            ]
        ], null, 2, 'div', [[
                'class',
                'mat-tab-body-content'
            ]
        ], [[
                24,
                '@translateTab',
                0
            ]
        ], [
            [
                null,
                '@translateTab.start'
            ],
            [
                null,
                '@translateTab.done'
            ]
        ], function (v, en, $event) {
            var ad = true;
            var co = v.component;
            if (('@translateTab.start' === en)) {
                var pd_0 = (co._onTranslateTabStarted($event) !== false);
                ad = (pd_0 && ad);
            }
            if (('@translateTab.done' === en)) {
                var pd_1 = (co._onTranslateTabComplete($event) !== false);
                ad = (pd_1 && ad);
            }
            return ad;
        }, null, null)),
        (l()(), import0.ɵand(8388608, null, null, 1, null, View_MdTabBody_1)),
        import0.ɵdid(73728, [[
                1,
                4
            ]
        ], 0, import2.PortalHostDirective, [
            import0.ComponentFactoryResolver,
            import0.ViewContainerRef
        ], { portal: [
                0,
                'portal'
            ]
        }, null)
    ], function (ck, v) {
        var currVal_1 = '';
        ck(v, 3, 0, currVal_1);
    }, function (ck, v) {
        var co = v.component;
        var currVal_0 = (co._canBeAnimated ? co._position : null);
        ck(v, 1, 0, currVal_0);
    });
}
function View_MdTabBody_Host_0(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 1, 'md-tab-body', [], [[
                2,
                'mat-tab-body',
                null
            ]
        ], null, null, View_MdTabBody_0, RenderType_MdTabBody)),
        import0.ɵdid(5300224, null, 0, import1.MdTabBody, [
            [
                2,
                import3.Dir
            ],
            import0.ElementRef,
            import0.ChangeDetectorRef
        ], null, null)
    ], function (ck, v) {
        ck(v, 1, 0);
    }, function (ck, v) {
        var currVal_0 = true;
        ck(v, 0, 0, currVal_0);
    });
}
export var MdTabBodyNgFactory = import0.ɵccf('md-tab-body, mat-tab-body', import1.MdTabBody, View_MdTabBody_Host_0, {
    _content: 'content',
    position: 'position',
    origin: 'origin'
}, {
    onCentering: 'onCentering',
    onCentered: 'onCentered'
}, []);
