// @angular/material/core/option/option.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from '@angular/core';
import * as import1 from '@angular/material/core/option/option';
import * as import2 from '@angular/material/core/compatibility/compatibility';
import * as import3 from '@angular/material/core/ripple/ripple';
import * as import4 from '@angular/common';
import * as import5 from '@angular/material/core/overlay/scroll/scroll-dispatcher';
import * as import6 from '@angular/material/core/overlay/position/viewport-ruler';
class MdOptionModuleInjector extends import0.ɵNgModuleInjector<import1.MdOptionModule> {
  _CompatibilityModule_0:import2.CompatibilityModule;
  _MdRippleModule_1:import3.MdRippleModule;
  _CommonModule_2:import4.CommonModule;
  _MdOptionModule_3:import1.MdOptionModule;
  __ScrollDispatcher_4:any;
  __ViewportRuler_5:any;
  __NgLocalization_6:import4.NgLocaleLocalization;
  constructor(parent:import0.Injector) {
    super(parent,([] as any[]),([] as any[]));
  }
  get _ScrollDispatcher_4():any {
    if ((this.__ScrollDispatcher_4 == null)) { (this.__ScrollDispatcher_4 = import5.SCROLL_DISPATCHER_PROVIDER_FACTORY(this.parent.get(import5.ScrollDispatcher,(null as any)))); }
    return this.__ScrollDispatcher_4;
  }
  get _ViewportRuler_5():any {
    if ((this.__ViewportRuler_5 == null)) { (this.__ViewportRuler_5 = import6.VIEWPORT_RULER_PROVIDER_FACTORY(this.parent.get(import6.ViewportRuler,(null as any)),this._ScrollDispatcher_4)); }
    return this.__ViewportRuler_5;
  }
  get _NgLocalization_6():import4.NgLocaleLocalization {
    if ((this.__NgLocalization_6 == null)) { (this.__NgLocalization_6 = new import4.NgLocaleLocalization(this.parent.get(import0.LOCALE_ID))); }
    return this.__NgLocalization_6;
  }
  createInternal():import1.MdOptionModule {
    this._CompatibilityModule_0 = new import2.CompatibilityModule();
    this._MdRippleModule_1 = new import3.MdRippleModule();
    this._CommonModule_2 = new import4.CommonModule();
    this._MdOptionModule_3 = new import1.MdOptionModule();
    return this._MdOptionModule_3;
  }
  getInternal(token:any,notFoundResult:any):any {
    if ((token === import2.CompatibilityModule)) { return this._CompatibilityModule_0; }
    if ((token === import3.MdRippleModule)) { return this._MdRippleModule_1; }
    if ((token === import4.CommonModule)) { return this._CommonModule_2; }
    if ((token === import1.MdOptionModule)) { return this._MdOptionModule_3; }
    if ((token === import5.ScrollDispatcher)) { return this._ScrollDispatcher_4; }
    if ((token === import6.ViewportRuler)) { return this._ViewportRuler_5; }
    if ((token === import4.NgLocalization)) { return this._NgLocalization_6; }
    return notFoundResult;
  }
  destroyInternal():void {
  }
}
export const MdOptionModuleNgFactory:import0.NgModuleFactory<import1.MdOptionModule> = new import0.NgModuleFactory<any>(MdOptionModuleInjector,import1.MdOptionModule);
const styles_MdOption:any[] = ([] as any[]);
export const RenderType_MdOption:import0.RendererType2 = import0.ɵcrt({
  encapsulation: 2,
  styles: styles_MdOption,
  data: {}
}
);
function View_MdOption_1(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    (l()(),import0.ɵeld(0,(null as any),(null as any),1,'div',[
      [
        'class',
        'mat-option-ripple'
      ]
      ,
      [
        'md-ripple',
        ''
      ]

    ]
    ,[
      [
        2,
        'mat-ripple',
        (null as any)
      ]
      ,
      [
        2,
        'mat-ripple-unbounded',
        (null as any)
      ]

    ]
    ,(null as any),(null as any),(null as any),(null as any))),
    import0.ɵdid(335872,(null as any),0,import3.MdRipple,[
      import0.ElementRef,
      import0.NgZone,
      import6.ViewportRuler
    ]
      ,{trigger: [
        0,
        'trigger'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_2:any = co._getHostElement();
    ck(v,1,0,currVal_2);
  },(ck,v) => {
    const currVal_0:any = true;
    const currVal_1:any = import0.ɵnov(v,1).unbounded;
    ck(v,0,0,currVal_0,currVal_1);
  });
}
export function View_MdOption_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
    import0.ɵncd((null as any),0),
    (l()(),import0.ɵand(8388608,(null as any),(null as any),1,(null as any),View_MdOption_1)),
    import0.ɵdid(8192,(null as any),0,import4.NgIf,[
      import0.ViewContainerRef,
      import0.TemplateRef
    ]
      ,{ngIf: [
        0,
        'ngIf'
      ]
    },(null as any))
  ]
  ,(ck,v) => {
    var co:import1.MdOption = v.component;
    const currVal_0:boolean = !co.disabled;
    ck(v,2,0,currVal_0);
  },(null as any));
}
function View_MdOption_Host_0(l:any):import0.ɵViewDefinition {
  return import0.ɵvid(0,[
      (l()(),import0.ɵeld(0,(null as any),(null as any),1,'md-option',[[
        'role',
        'option'
      ]
    ],[
      [
        1,
        'tabindex',
        0
      ]
      ,
      [
        2,
        'mat-selected',
        (null as any)
      ]
      ,
      [
        2,
        'mat-active',
        (null as any)
      ]
      ,
      [
        8,
        'id',
        0
      ]
      ,
      [
        1,
        'aria-selected',
        0
      ]
      ,
      [
        1,
        'aria-disabled',
        0
      ]
      ,
      [
        2,
        'mat-option-disabled',
        (null as any)
      ]
      ,
      [
        2,
        'mat-option',
        (null as any)
      ]

    ]
    ,[
      [
        (null as any),
        'click'
      ]
      ,
      [
        (null as any),
        'keydown'
      ]

    ]
    ,(v,en,$event) => {
      var ad:boolean = true;
      if (('click' === en)) {
        const pd_0:any = ((<any>import0.ɵnov(v,1)._selectViaInteraction()) !== false);
        ad = (pd_0 && ad);
      }
      if (('keydown' === en)) {
        const pd_1:any = ((<any>import0.ɵnov(v,1)._handleKeydown($event)) !== false);
        ad = (pd_1 && ad);
      }
      return ad;
    },View_MdOption_0,RenderType_MdOption)),
    import0.ɵdid(24576,(null as any),0,import1.MdOption,[
      import0.ElementRef,
      import0.Renderer
    ]
    ,(null as any),(null as any))
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = import0.ɵnov(v,1)._getTabIndex();
    const currVal_1:any = import0.ɵnov(v,1).selected;
    const currVal_2:any = import0.ɵnov(v,1).active;
    const currVal_3:any = import0.ɵnov(v,1).id;
    const currVal_4:any = import0.ɵnov(v,1).selected.toString();
    const currVal_5:any = import0.ɵnov(v,1).disabled.toString();
    const currVal_6:any = import0.ɵnov(v,1).disabled;
    const currVal_7:any = true;
    ck(v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7);
  });
}
export const MdOptionNgFactory:import0.ComponentFactory<import1.MdOption> = import0.ɵccf('md-option, mat-option',import1.MdOption,View_MdOption_Host_0,{
  value: 'value',
  disabled: 'disabled'
}
,{onSelect: 'onSelect'},['*']);
