// @angular/material/autocomplete/autocomplete.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */


import * as import0 from './autocomplete.css.ngstyle';
import * as import1 from '@angular/core';
import * as import2 from '@angular/common';
import * as import3 from '@angular/material/autocomplete/autocomplete';
const styles_MdAutocomplete:any[] = [import0.styles];
export const RenderType_MdAutocomplete:import1.RendererType2 = import1.ɵcrt({
  encapsulation: 2,
  styles: styles_MdAutocomplete,
  data: {}
}
);
function View_MdAutocomplete_1(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
    (l()(),import1.ɵeld(0,[
      [
        2,
        0
      ]
      ,
      [
        'panel',
        1
      ]

    ]
    ,(null as any),2,'div',[
      [
        'class',
        'mat-autocomplete-panel'
      ]
      ,
      [
        'role',
        'listbox'
      ]

    ]
      ,[[
        8,
        'id',
        0
      ]
    ],(null as any),(null as any),(null as any),(null as any))),
    import1.ɵdid(139264,(null as any),0,import2.NgClass,[
      import1.IterableDiffers,
      import1.KeyValueDiffers,
      import1.ElementRef,
      import1.Renderer
    ]
    ,{
      klass: [
        0,
        'klass'
      ]
      ,
      ngClass: [
        1,
        'ngClass'
      ]

    }
    ,(null as any)),
    import1.ɵncd((null as any),0)
  ]
  ,(ck,v) => {
    var co:any = v.component;
    const currVal_1:any = 'mat-autocomplete-panel';
    const currVal_2:any = co._getClassList();
    ck(v,1,0,currVal_1,currVal_2);
  },(ck,v) => {
    var co:any = v.component;
    const currVal_0:any = co.id;
    ck(v,0,0,currVal_0);
  });
}
export function View_MdAutocomplete_0(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
    import1.ɵqud(201326592,1,{template: 0}),
    import1.ɵqud(335544320,2,{panel: 0}),
      (l()(),import1.ɵand(0,[[
        1,
        2
      ]
    ],(null as any),0,(null as any),View_MdAutocomplete_1))
  ]
  ,(null as any),(null as any));
}
function View_MdAutocomplete_Host_0(l:any):import1.ɵViewDefinition {
  return import1.ɵvid(0,[
      (l()(),import1.ɵeld(0,(null as any),(null as any),2,'md-autocomplete',([] as any[]),[[
        2,
        'mat-autocomplete',
        (null as any)
      ]
    ],(null as any),(null as any),View_MdAutocomplete_0,RenderType_MdAutocomplete)),
    import1.ɵdid(24576,(null as any),1,import3.MdAutocomplete,([] as any[]),(null as any),(null as any)),
    import1.ɵqud(301989888,1,{options: 1})
  ]
  ,(null as any),(ck,v) => {
    const currVal_0:any = true;
    ck(v,0,0,currVal_0);
  });
}
export const MdAutocompleteNgFactory:import1.ComponentFactory<import3.MdAutocomplete> = import1.ɵccf('md-autocomplete, mat-autocomplete',import3.MdAutocomplete,View_MdAutocomplete_Host_0,{displayWith: 'displayWith'},{},['*']);
