// @angular/platform-browser/animations/index.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
/* tslint:disable */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
export { ɵWebAnimationsStyleNormalizer as ɵWebAnimationsStyleNormalizer_7, AnimationDriver as AnimationDriver_10, ɵAnimationStyleNormalizer as ɵAnimationStyleNormalizer_11, ɵAnimationEngine as ɵAnimationEngine_12, ɵNoopAnimationEngine as ɵNoopAnimationEngine_17 } from '@angular/animations/browser';
import * as import0 from '@angular/core';
import * as import1 from '@angular/platform-browser/animations';
import * as import2 from '@angular/common';
import * as import3 from '@angular/platform-browser';
import * as import4 from '@angular/animations/browser';
var BrowserAnimationsModuleInjector = (function (_super) {
    __extends(BrowserAnimationsModuleInjector, _super);
    function BrowserAnimationsModuleInjector(parent) {
        return _super.call(this, parent, [], []) || this;
    }
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_LOCALE_ID_9", {
        get: function () {
            if ((this.__LOCALE_ID_9 == null)) {
                (this.__LOCALE_ID_9 = import0.ɵn(this.parent.get(import0.LOCALE_ID, null)));
            }
            return this.__LOCALE_ID_9;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_NgLocalization_10", {
        get: function () {
            if ((this.__NgLocalization_10 == null)) {
                (this.__NgLocalization_10 = new import2.NgLocaleLocalization(this._LOCALE_ID_9));
            }
            return this.__NgLocalization_10;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_Compiler_11", {
        get: function () {
            if ((this.__Compiler_11 == null)) {
                (this.__Compiler_11 = new import0.Compiler());
            }
            return this.__Compiler_11;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_APP_ID_12", {
        get: function () {
            if ((this.__APP_ID_12 == null)) {
                (this.__APP_ID_12 = import0.ɵg());
            }
            return this.__APP_ID_12;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_IterableDiffers_13", {
        get: function () {
            if ((this.__IterableDiffers_13 == null)) {
                (this.__IterableDiffers_13 = import0.ɵl());
            }
            return this.__IterableDiffers_13;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_KeyValueDiffers_14", {
        get: function () {
            if ((this.__KeyValueDiffers_14 == null)) {
                (this.__KeyValueDiffers_14 = import0.ɵm());
            }
            return this.__KeyValueDiffers_14;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_DomSanitizer_15", {
        get: function () {
            if ((this.__DomSanitizer_15 == null)) {
                (this.__DomSanitizer_15 = new import3.ɵe(this.parent.get(import3.DOCUMENT)));
            }
            return this.__DomSanitizer_15;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_Sanitizer_16", {
        get: function () {
            if ((this.__Sanitizer_16 == null)) {
                (this.__Sanitizer_16 = this._DomSanitizer_15);
            }
            return this.__Sanitizer_16;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_HAMMER_GESTURE_CONFIG_17", {
        get: function () {
            if ((this.__HAMMER_GESTURE_CONFIG_17 == null)) {
                (this.__HAMMER_GESTURE_CONFIG_17 = new import3.HammerGestureConfig());
            }
            return this.__HAMMER_GESTURE_CONFIG_17;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_EVENT_MANAGER_PLUGINS_18", {
        get: function () {
            if ((this.__EVENT_MANAGER_PLUGINS_18 == null)) {
                (this.__EVENT_MANAGER_PLUGINS_18 = [
                    new import3.ɵDomEventsPlugin(this.parent.get(import3.DOCUMENT)),
                    new import3.ɵKeyEventsPlugin(this.parent.get(import3.DOCUMENT)),
                    new import3.ɵHammerGesturesPlugin(this.parent.get(import3.DOCUMENT), this._HAMMER_GESTURE_CONFIG_17)
                ]);
            }
            return this.__EVENT_MANAGER_PLUGINS_18;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_EventManager_19", {
        get: function () {
            if ((this.__EventManager_19 == null)) {
                (this.__EventManager_19 = new import3.EventManager(this._EVENT_MANAGER_PLUGINS_18, this.parent.get(import0.NgZone)));
            }
            return this.__EventManager_19;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_\u0275DomSharedStylesHost_20", {
        get: function () {
            if ((this.__ɵDomSharedStylesHost_20 == null)) {
                (this.__ɵDomSharedStylesHost_20 = new import3.ɵDomSharedStylesHost(this.parent.get(import3.DOCUMENT)));
            }
            return this.__ɵDomSharedStylesHost_20;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_\u0275DomRendererFactory2_21", {
        get: function () {
            if ((this.__ɵDomRendererFactory2_21 == null)) {
                (this.__ɵDomRendererFactory2_21 = new import3.ɵDomRendererFactory2(this._EventManager_19, this._ɵDomSharedStylesHost_20));
            }
            return this.__ɵDomRendererFactory2_21;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_AnimationDriver_22", {
        get: function () {
            if ((this.__AnimationDriver_22 == null)) {
                (this.__AnimationDriver_22 = import1.ɵb());
            }
            return this.__AnimationDriver_22;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_\u0275AnimationStyleNormalizer_23", {
        get: function () {
            if ((this.__ɵAnimationStyleNormalizer_23 == null)) {
                (this.__ɵAnimationStyleNormalizer_23 = import1.ɵc());
            }
            return this.__ɵAnimationStyleNormalizer_23;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_\u0275AnimationEngine_24", {
        get: function () {
            if ((this.__ɵAnimationEngine_24 == null)) {
                (this.__ɵAnimationEngine_24 = new import1.ɵa(this._AnimationDriver_22, this._ɵAnimationStyleNormalizer_23));
            }
            return this.__ɵAnimationEngine_24;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_RendererFactory2_25", {
        get: function () {
            if ((this.__RendererFactory2_25 == null)) {
                (this.__RendererFactory2_25 = import1.ɵd(this._ɵDomRendererFactory2_21, this._ɵAnimationEngine_24, this.parent.get(import0.NgZone)));
            }
            return this.__RendererFactory2_25;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_\u0275SharedStylesHost_26", {
        get: function () {
            if ((this.__ɵSharedStylesHost_26 == null)) {
                (this.__ɵSharedStylesHost_26 = this._ɵDomSharedStylesHost_20);
            }
            return this.__ɵSharedStylesHost_26;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_Testability_27", {
        get: function () {
            if ((this.__Testability_27 == null)) {
                (this.__Testability_27 = new import0.Testability(this.parent.get(import0.NgZone)));
            }
            return this.__Testability_27;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_Meta_28", {
        get: function () {
            if ((this.__Meta_28 == null)) {
                (this.__Meta_28 = new import3.Meta(this.parent.get(import3.DOCUMENT)));
            }
            return this.__Meta_28;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(BrowserAnimationsModuleInjector.prototype, "_Title_29", {
        get: function () {
            if ((this.__Title_29 == null)) {
                (this.__Title_29 = new import3.Title(this.parent.get(import3.DOCUMENT)));
            }
            return this.__Title_29;
        },
        enumerable: true,
        configurable: true
    });
    BrowserAnimationsModuleInjector.prototype.createInternal = function () {
        this._CommonModule_0 = new import2.CommonModule();
        this._ErrorHandler_1 = import3.ɵa();
        this._APP_INITIALIZER_2 = [
            import0.ɵo,
            import3.ɵc(this.parent.get(import3.NgProbeToken, null), this.parent.get(import0.NgProbeToken, null))
        ];
        this._ApplicationInitStatus_3 = new import0.ApplicationInitStatus(this._APP_INITIALIZER_2);
        this._ɵf_4 = new import0.ɵf(this.parent.get(import0.NgZone), this.parent.get(import0.ɵConsole), this, this._ErrorHandler_1, this.componentFactoryResolver, this._ApplicationInitStatus_3);
        this._ApplicationRef_5 = this._ɵf_4;
        this._ApplicationModule_6 = new import0.ApplicationModule(this._ApplicationRef_5);
        this._BrowserModule_7 = new import3.BrowserModule(this.parent.get(import3.BrowserModule, null));
        this._BrowserAnimationsModule_8 = new import1.BrowserAnimationsModule();
        return this._BrowserAnimationsModule_8;
    };
    BrowserAnimationsModuleInjector.prototype.getInternal = function (token, notFoundResult) {
        if ((token === import2.CommonModule)) {
            return this._CommonModule_0;
        }
        if ((token === import0.ErrorHandler)) {
            return this._ErrorHandler_1;
        }
        if ((token === import0.APP_INITIALIZER)) {
            return this._APP_INITIALIZER_2;
        }
        if ((token === import0.ApplicationInitStatus)) {
            return this._ApplicationInitStatus_3;
        }
        if ((token === import0.ɵf)) {
            return this._ɵf_4;
        }
        if ((token === import0.ApplicationRef)) {
            return this._ApplicationRef_5;
        }
        if ((token === import0.ApplicationModule)) {
            return this._ApplicationModule_6;
        }
        if ((token === import3.BrowserModule)) {
            return this._BrowserModule_7;
        }
        if ((token === import1.BrowserAnimationsModule)) {
            return this._BrowserAnimationsModule_8;
        }
        if ((token === import0.LOCALE_ID)) {
            return this._LOCALE_ID_9;
        }
        if ((token === import2.NgLocalization)) {
            return this._NgLocalization_10;
        }
        if ((token === import0.Compiler)) {
            return this._Compiler_11;
        }
        if ((token === import0.APP_ID)) {
            return this._APP_ID_12;
        }
        if ((token === import0.IterableDiffers)) {
            return this._IterableDiffers_13;
        }
        if ((token === import0.KeyValueDiffers)) {
            return this._KeyValueDiffers_14;
        }
        if ((token === import3.DomSanitizer)) {
            return this._DomSanitizer_15;
        }
        if ((token === import0.Sanitizer)) {
            return this._Sanitizer_16;
        }
        if ((token === import3.HAMMER_GESTURE_CONFIG)) {
            return this._HAMMER_GESTURE_CONFIG_17;
        }
        if ((token === import3.EVENT_MANAGER_PLUGINS)) {
            return this._EVENT_MANAGER_PLUGINS_18;
        }
        if ((token === import3.EventManager)) {
            return this._EventManager_19;
        }
        if ((token === import3.ɵDomSharedStylesHost)) {
            return this._ɵDomSharedStylesHost_20;
        }
        if ((token === import3.ɵDomRendererFactory2)) {
            return this._ɵDomRendererFactory2_21;
        }
        if ((token === import4.AnimationDriver)) {
            return this._AnimationDriver_22;
        }
        if ((token === import4.ɵAnimationStyleNormalizer)) {
            return this._ɵAnimationStyleNormalizer_23;
        }
        if ((token === import4.ɵAnimationEngine)) {
            return this._ɵAnimationEngine_24;
        }
        if ((token === import0.RendererFactory2)) {
            return this._RendererFactory2_25;
        }
        if ((token === import3.ɵSharedStylesHost)) {
            return this._ɵSharedStylesHost_26;
        }
        if ((token === import0.Testability)) {
            return this._Testability_27;
        }
        if ((token === import3.Meta)) {
            return this._Meta_28;
        }
        if ((token === import3.Title)) {
            return this._Title_29;
        }
        return notFoundResult;
    };
    BrowserAnimationsModuleInjector.prototype.destroyInternal = function () {
        this._ɵf_4.ngOnDestroy();
        (this.__ɵDomSharedStylesHost_20 && this._ɵDomSharedStylesHost_20.ngOnDestroy());
    };
    return BrowserAnimationsModuleInjector;
}(import0.ɵNgModuleInjector));
export var BrowserAnimationsModuleNgFactory = new import0.NgModuleFactory(BrowserAnimationsModuleInjector, import1.BrowserAnimationsModule);
var NoopAnimationsModuleInjector = (function (_super) {
    __extends(NoopAnimationsModuleInjector, _super);
    function NoopAnimationsModuleInjector(parent) {
        return _super.call(this, parent, [], []) || this;
    }
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_LOCALE_ID_9", {
        get: function () {
            if ((this.__LOCALE_ID_9 == null)) {
                (this.__LOCALE_ID_9 = import0.ɵn(this.parent.get(import0.LOCALE_ID, null)));
            }
            return this.__LOCALE_ID_9;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_NgLocalization_10", {
        get: function () {
            if ((this.__NgLocalization_10 == null)) {
                (this.__NgLocalization_10 = new import2.NgLocaleLocalization(this._LOCALE_ID_9));
            }
            return this.__NgLocalization_10;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_Compiler_11", {
        get: function () {
            if ((this.__Compiler_11 == null)) {
                (this.__Compiler_11 = new import0.Compiler());
            }
            return this.__Compiler_11;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_APP_ID_12", {
        get: function () {
            if ((this.__APP_ID_12 == null)) {
                (this.__APP_ID_12 = import0.ɵg());
            }
            return this.__APP_ID_12;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_IterableDiffers_13", {
        get: function () {
            if ((this.__IterableDiffers_13 == null)) {
                (this.__IterableDiffers_13 = import0.ɵl());
            }
            return this.__IterableDiffers_13;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_KeyValueDiffers_14", {
        get: function () {
            if ((this.__KeyValueDiffers_14 == null)) {
                (this.__KeyValueDiffers_14 = import0.ɵm());
            }
            return this.__KeyValueDiffers_14;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_DomSanitizer_15", {
        get: function () {
            if ((this.__DomSanitizer_15 == null)) {
                (this.__DomSanitizer_15 = new import3.ɵe(this.parent.get(import3.DOCUMENT)));
            }
            return this.__DomSanitizer_15;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_Sanitizer_16", {
        get: function () {
            if ((this.__Sanitizer_16 == null)) {
                (this.__Sanitizer_16 = this._DomSanitizer_15);
            }
            return this.__Sanitizer_16;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_HAMMER_GESTURE_CONFIG_17", {
        get: function () {
            if ((this.__HAMMER_GESTURE_CONFIG_17 == null)) {
                (this.__HAMMER_GESTURE_CONFIG_17 = new import3.HammerGestureConfig());
            }
            return this.__HAMMER_GESTURE_CONFIG_17;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_EVENT_MANAGER_PLUGINS_18", {
        get: function () {
            if ((this.__EVENT_MANAGER_PLUGINS_18 == null)) {
                (this.__EVENT_MANAGER_PLUGINS_18 = [
                    new import3.ɵDomEventsPlugin(this.parent.get(import3.DOCUMENT)),
                    new import3.ɵKeyEventsPlugin(this.parent.get(import3.DOCUMENT)),
                    new import3.ɵHammerGesturesPlugin(this.parent.get(import3.DOCUMENT), this._HAMMER_GESTURE_CONFIG_17)
                ]);
            }
            return this.__EVENT_MANAGER_PLUGINS_18;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_EventManager_19", {
        get: function () {
            if ((this.__EventManager_19 == null)) {
                (this.__EventManager_19 = new import3.EventManager(this._EVENT_MANAGER_PLUGINS_18, this.parent.get(import0.NgZone)));
            }
            return this.__EventManager_19;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_\u0275DomSharedStylesHost_20", {
        get: function () {
            if ((this.__ɵDomSharedStylesHost_20 == null)) {
                (this.__ɵDomSharedStylesHost_20 = new import3.ɵDomSharedStylesHost(this.parent.get(import3.DOCUMENT)));
            }
            return this.__ɵDomSharedStylesHost_20;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_\u0275DomRendererFactory2_21", {
        get: function () {
            if ((this.__ɵDomRendererFactory2_21 == null)) {
                (this.__ɵDomRendererFactory2_21 = new import3.ɵDomRendererFactory2(this._EventManager_19, this._ɵDomSharedStylesHost_20));
            }
            return this.__ɵDomRendererFactory2_21;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_\u0275AnimationEngine_22", {
        get: function () {
            if ((this.__ɵAnimationEngine_22 == null)) {
                (this.__ɵAnimationEngine_22 = new import4.ɵNoopAnimationEngine());
            }
            return this.__ɵAnimationEngine_22;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_RendererFactory2_23", {
        get: function () {
            if ((this.__RendererFactory2_23 == null)) {
                (this.__RendererFactory2_23 = import1.ɵd(this._ɵDomRendererFactory2_21, this._ɵAnimationEngine_22, this.parent.get(import0.NgZone)));
            }
            return this.__RendererFactory2_23;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_\u0275SharedStylesHost_24", {
        get: function () {
            if ((this.__ɵSharedStylesHost_24 == null)) {
                (this.__ɵSharedStylesHost_24 = this._ɵDomSharedStylesHost_20);
            }
            return this.__ɵSharedStylesHost_24;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_Testability_25", {
        get: function () {
            if ((this.__Testability_25 == null)) {
                (this.__Testability_25 = new import0.Testability(this.parent.get(import0.NgZone)));
            }
            return this.__Testability_25;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_Meta_26", {
        get: function () {
            if ((this.__Meta_26 == null)) {
                (this.__Meta_26 = new import3.Meta(this.parent.get(import3.DOCUMENT)));
            }
            return this.__Meta_26;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NoopAnimationsModuleInjector.prototype, "_Title_27", {
        get: function () {
            if ((this.__Title_27 == null)) {
                (this.__Title_27 = new import3.Title(this.parent.get(import3.DOCUMENT)));
            }
            return this.__Title_27;
        },
        enumerable: true,
        configurable: true
    });
    NoopAnimationsModuleInjector.prototype.createInternal = function () {
        this._CommonModule_0 = new import2.CommonModule();
        this._ErrorHandler_1 = import3.ɵa();
        this._APP_INITIALIZER_2 = [
            import0.ɵo,
            import3.ɵc(this.parent.get(import3.NgProbeToken, null), this.parent.get(import0.NgProbeToken, null))
        ];
        this._ApplicationInitStatus_3 = new import0.ApplicationInitStatus(this._APP_INITIALIZER_2);
        this._ɵf_4 = new import0.ɵf(this.parent.get(import0.NgZone), this.parent.get(import0.ɵConsole), this, this._ErrorHandler_1, this.componentFactoryResolver, this._ApplicationInitStatus_3);
        this._ApplicationRef_5 = this._ɵf_4;
        this._ApplicationModule_6 = new import0.ApplicationModule(this._ApplicationRef_5);
        this._BrowserModule_7 = new import3.BrowserModule(this.parent.get(import3.BrowserModule, null));
        this._NoopAnimationsModule_8 = new import1.NoopAnimationsModule();
        return this._NoopAnimationsModule_8;
    };
    NoopAnimationsModuleInjector.prototype.getInternal = function (token, notFoundResult) {
        if ((token === import2.CommonModule)) {
            return this._CommonModule_0;
        }
        if ((token === import0.ErrorHandler)) {
            return this._ErrorHandler_1;
        }
        if ((token === import0.APP_INITIALIZER)) {
            return this._APP_INITIALIZER_2;
        }
        if ((token === import0.ApplicationInitStatus)) {
            return this._ApplicationInitStatus_3;
        }
        if ((token === import0.ɵf)) {
            return this._ɵf_4;
        }
        if ((token === import0.ApplicationRef)) {
            return this._ApplicationRef_5;
        }
        if ((token === import0.ApplicationModule)) {
            return this._ApplicationModule_6;
        }
        if ((token === import3.BrowserModule)) {
            return this._BrowserModule_7;
        }
        if ((token === import1.NoopAnimationsModule)) {
            return this._NoopAnimationsModule_8;
        }
        if ((token === import0.LOCALE_ID)) {
            return this._LOCALE_ID_9;
        }
        if ((token === import2.NgLocalization)) {
            return this._NgLocalization_10;
        }
        if ((token === import0.Compiler)) {
            return this._Compiler_11;
        }
        if ((token === import0.APP_ID)) {
            return this._APP_ID_12;
        }
        if ((token === import0.IterableDiffers)) {
            return this._IterableDiffers_13;
        }
        if ((token === import0.KeyValueDiffers)) {
            return this._KeyValueDiffers_14;
        }
        if ((token === import3.DomSanitizer)) {
            return this._DomSanitizer_15;
        }
        if ((token === import0.Sanitizer)) {
            return this._Sanitizer_16;
        }
        if ((token === import3.HAMMER_GESTURE_CONFIG)) {
            return this._HAMMER_GESTURE_CONFIG_17;
        }
        if ((token === import3.EVENT_MANAGER_PLUGINS)) {
            return this._EVENT_MANAGER_PLUGINS_18;
        }
        if ((token === import3.EventManager)) {
            return this._EventManager_19;
        }
        if ((token === import3.ɵDomSharedStylesHost)) {
            return this._ɵDomSharedStylesHost_20;
        }
        if ((token === import3.ɵDomRendererFactory2)) {
            return this._ɵDomRendererFactory2_21;
        }
        if ((token === import4.ɵAnimationEngine)) {
            return this._ɵAnimationEngine_22;
        }
        if ((token === import0.RendererFactory2)) {
            return this._RendererFactory2_23;
        }
        if ((token === import3.ɵSharedStylesHost)) {
            return this._ɵSharedStylesHost_24;
        }
        if ((token === import0.Testability)) {
            return this._Testability_25;
        }
        if ((token === import3.Meta)) {
            return this._Meta_26;
        }
        if ((token === import3.Title)) {
            return this._Title_27;
        }
        return notFoundResult;
    };
    NoopAnimationsModuleInjector.prototype.destroyInternal = function () {
        this._ɵf_4.ngOnDestroy();
        (this.__ɵDomSharedStylesHost_20 && this._ɵDomSharedStylesHost_20.ngOnDestroy());
    };
    return NoopAnimationsModuleInjector;
}(import0.ɵNgModuleInjector));
export var NoopAnimationsModuleNgFactory = new import0.NgModuleFactory(NoopAnimationsModuleInjector, import1.NoopAnimationsModule);
